<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Stephanie C. Hicks, PhD - Projects</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Stephanie C. Hicks, PhD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./people.html" rel="" target="">
 <span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./projects.html" rel="" target="" aria-current="page">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./software.html" rel="" target="">
 <span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./teaching.html" rel="" target="">
 <span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./talks.html" rel="" target="">
 <span class="menu-text">Talks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./blog.html" rel="" target="">
 <span class="menu-text">Blog</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./join.html" rel="" target="">
 <span class="menu-text">Join us</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#reproducible-research" id="toc-reproducible-research" class="nav-link active" data-scroll-target="#reproducible-research">Reproducible research</a></li>
  <li><a href="#high-throughput-biological-data" id="toc-high-throughput-biological-data" class="nav-link" data-scroll-target="#high-throughput-biological-data">High-throughput biological data</a>
  <ul class="collapse">
  <li><a href="#methods-for-normalizing-and-removing-biases" id="toc-methods-for-normalizing-and-removing-biases" class="nav-link" data-scroll-target="#methods-for-normalizing-and-removing-biases">Methods for normalizing and removing biases</a></li>
  <li><a href="#methods-for-deconvolution-bulk-tissue-samples" id="toc-methods-for-deconvolution-bulk-tissue-samples" class="nav-link" data-scroll-target="#methods-for-deconvolution-bulk-tissue-samples">Methods for deconvolution bulk tissue samples</a></li>
  <li><a href="#methods-to-identify-differences-between-groups" id="toc-methods-to-identify-differences-between-groups" class="nav-link" data-scroll-target="#methods-to-identify-differences-between-groups">Methods to identify differences between groups</a></li>
  </ul></li>
  <li><a href="#genomics-data-from-single-cells" id="toc-genomics-data-from-single-cells" class="nav-link" data-scroll-target="#genomics-data-from-single-cells">Genomics data from single cells</a>
  <ul class="collapse">
  <li><a href="#methods-to-identify-and-remove-systematic-errors-and-biases" id="toc-methods-to-identify-and-remove-systematic-errors-and-biases" class="nav-link" data-scroll-target="#methods-to-identify-and-remove-systematic-errors-and-biases">Methods to identify and remove systematic errors and biases</a></li>
  <li><a href="#methods-for-distance-metrics-and-unsupervised-analyses" id="toc-methods-for-distance-metrics-and-unsupervised-analyses" class="nav-link" data-scroll-target="#methods-for-distance-metrics-and-unsupervised-analyses">Methods for distance metrics and unsupervised analyses</a></li>
  <li><a href="#methods-to-identify-differences-between-groups-1" id="toc-methods-to-identify-differences-between-groups-1" class="nav-link" data-scroll-target="#methods-to-identify-differences-between-groups-1">Methods to identify differences between groups</a></li>
  <li><a href="#single-cell-applications" id="toc-single-cell-applications" class="nav-link" data-scroll-target="#single-cell-applications">Single cell applications</a>
  <ul class="collapse">
  <li><a href="#high-grade-serous-ovarian-cancer" id="toc-high-grade-serous-ovarian-cancer" class="nav-link" data-scroll-target="#high-grade-serous-ovarian-cancer">High-grade serous ovarian cancer</a></li>
  <li><a href="#neuroscience" id="toc-neuroscience" class="nav-link" data-scroll-target="#neuroscience">Neuroscience</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#genomics-data-with-spatial-resolution" id="toc-genomics-data-with-spatial-resolution" class="nav-link" data-scroll-target="#genomics-data-with-spatial-resolution">Genomics data with spatial resolution</a>
  <ul class="collapse">
  <li><a href="#data-storage-and-data-visualization" id="toc-data-storage-and-data-visualization" class="nav-link" data-scroll-target="#data-storage-and-data-visualization">Data storage and data visualization</a></li>
  <li><a href="#methods-to-identify-spatially-variable-genes" id="toc-methods-to-identify-spatially-variable-genes" class="nav-link" data-scroll-target="#methods-to-identify-spatially-variable-genes">Methods to identify spatially variable genes</a></li>
  <li><a href="#spatial-atlases" id="toc-spatial-atlases" class="nav-link" data-scroll-target="#spatial-atlases">Spatial atlases</a>
  <ul class="collapse">
  <li><a href="#neuroscience-1" id="toc-neuroscience-1" class="nav-link" data-scroll-target="#neuroscience-1">Neuroscience</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#benchmarking-open-source-software-for-genomics" id="toc-benchmarking-open-source-software-for-genomics" class="nav-link" data-scroll-target="#benchmarking-open-source-software-for-genomics">Benchmarking open-source software for genomics</a></li>
  <li><a href="#data-science-education" id="toc-data-science-education" class="nav-link" data-scroll-target="#data-science-education">Data science education</a>
  <ul class="collapse">
  <li><a href="#open-case-studies" id="toc-open-case-studies" class="nav-link" data-scroll-target="#open-case-studies">Open Case Studies</a></li>
  <li><a href="#analytic-design-theory" id="toc-analytic-design-theory" class="nav-link" data-scroll-target="#analytic-design-theory">Analytic Design Theory</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Projects</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="reproducible-research" class="level1">
<h1>Reproducible research</h1>
<p>With the advent of large-scale and high-throughput data collection coupled with the creation and implementation of complex statistical algorithms for data analysis, the reproducibility of modern data analyses, meaning the ability of independent analysts to recreate the results claimed by the original authors using the original data and analysis techniques, has become an important topic of discussion. I have contributed to developing reproducibility standards for machine learning in the life sciences (1), and also in public health (2). Furthermore, I am also an <a href="https://jasa-acs.github.io/repro-guide/">Associate Editor for Reproducibility</a> at the <a href="https://www.tandfonline.com/toc/uasa20/current">Journal of the American Statistical Association</a> (JASA) where I helped launch the <a href="https://jasa-acs.github.io/repro-award">JASA Reproducibility Award</a>.</p>
<ol type="1">
<li>Heil BJ, Hoffman MM, Markowetz F, Lee S-I, Greene CS, <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1038/s41592-021-01256-7">Reproducibility standards for machine learning in the life sciences</a>. <em>Nature Methods</em>.</li>
<li>Peng RD, <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1146/annurev-publhealth-012420-105110">Reproducible Research: A Retrospective</a>. <em>Annual Review of Public Health</em>.</li>
</ol>
</section>
<section id="high-throughput-biological-data" class="level1">
<h1>High-throughput biological data</h1>
<section id="methods-for-normalizing-and-removing-biases" class="level2">
<h2 class="anchored" data-anchor-id="methods-for-normalizing-and-removing-biases">Methods for normalizing and removing biases</h2>
<p>Normalization is an essential step for the analysis of genomics high-throughput data. Quantile normalization is one of the most widely used multi-sample normalization tools for applications including genotyping arrays, RNA-Sequencing (RNA-Seq), DNA methylation (DNAm), ChIP-Sequencing (ChIP-Seq) and brain imaging. However, quantile normalization relies on assumptions about the data-generation process that are not appropriate in some contexts. To address this, I developed two statistical methods and open-source software for normalizing high-throughput data (1, 2). More recently, I have developed methods and software for correcting and removing compositional biases found in metagenomic data (3), and co-expression analysis of correlated genes (4).</p>
<ol type="1">
<li><strong>Hicks SC</strong>, Irizarry RA (2015). <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0679-0">quantro: a data-driven approach to guide the choice of an appropriate normalization method</a>. <em>Genome Biology</em>.</li>
<li><strong>Hicks SC</strong>, Okrah K, Paulson JN, Quackenbush J, Irizarry RA, Bravo HC (2018). <a href="https://academic.oup.com/biostatistics/article-abstract/19/2/185/3949169">Smooth quantile normalization</a>. <em>Biostatistics</em>.</li>
<li>Kumar MS, Slud EV, Okrah K, <strong>Hicks SC</strong>, Hannenhalli S, Corrada Bravo H (2018). <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-5160-5">Analysis and correction of compositional bias in sparse sequencing count data</a>. <em>BMC Genomics</em>.</li>
<li>Wang Y, <strong>Hicks SC</strong>, Hansen KD (2022). <a href="https://doi.org/10.1371/journal.pcbi.1009954">Addressing the mean-correlation relationship in co-expression analysis</a>. <em>PLOS Computational Biology</em>.</li>
</ol>
</section>
<section id="methods-for-deconvolution-bulk-tissue-samples" class="level2">
<h2 class="anchored" data-anchor-id="methods-for-deconvolution-bulk-tissue-samples">Methods for deconvolution bulk tissue samples</h2>
<p>Bulk tissue samples are often profiled due to their ease of access and low cost to sequence. A major challenge in the analysis of bulk tissue samples is they are typically made up of heterogenous cell types, for example whole blood or brain tissue. This intra-sample cellular heterogeneity is a source of variation than can sometimes be confounded with an outcome or phenotype of interest. If unaccounted for, false positives ensue. Previously, I developed statistical methods to estimate the cell type proportions in whole blood DNAm samples (1). Currently, I am developing <a href="https://reporter.nih.gov/search/qPVsXfCAiEWzTUirO5-Ppg/project-details/10138028#similar-Projects">cellular deconvolution methods for human brain RNA-seq data</a> (R01MH13183). One major source of variation in brain issue is how the cell sizes and cell abundance varies across cell types, which can lead to inaccurate estimates of cell type proportions in bulk RNA-seq (2). I recently developed a data-driven method to estimate the RNA abundance from cell types in heterogeneous tissues using sc/snRNA-seq and smFISH, which can be used for deconovolution (3).</p>
<ol type="1">
<li><p><strong>Hicks SC</strong>, Irizarry RA (2019). <a href="https://doi.org/10.1186/s13059-019-1827-8">methylCC: technology-independent estimation of cell type composition using differentially methylated regions</a>. <em>Genome Biology</em>.</p></li>
<li><p>Maden SK, Kwon SH, Huuki-Myers LA, Collado-Torres L, <strong>Hicks SC</strong>, Maynard KR. (2023). <a href="https://arxiv.org/abs/2305.06501">Challenges and opportunities to computationally deconvolve heterogeneous tissue with varying cell sizes using single cell RNA-sequencing datasets</a>. <em>arXiv</em>.</p></li>
<li><p>Huuki-Myers LA, Montgomery KD, Kwon SH, Page SC, <strong>Hicks SC</strong>, Maynard KR, Collado-Torres L. (2022). <a href="https://doi.org/10.1186/s13059-023-03066-w">Data-driven Identification of Total RNA Expression Genes (TREGs) for Estimation of RNA Abundance in Heterogeneous Cell Types</a>. <em>Genome Biology</em>.</p></li>
</ol>
</section>
<section id="methods-to-identify-differences-between-groups" class="level2">
<h2 class="anchored" data-anchor-id="methods-to-identify-differences-between-groups">Methods to identify differences between groups</h2>
<p>Identifying differences between biological groups or conditions in high-throughput data is a standard data analysis task. Recent effors have focused on identify differences between genes, but also transcripts. For example, differential transcript usage (DTU) has been shown to be important between cells, tissues, and biological systems. We developed a computational tool to quantify heterogeneity in transcript usage within a population of bulk RNA-seq samples and identifies predominant subgroups along with their distinctive sets of DTU events (1).</p>
<ol type="1">
<li>Erdogdu B, Varabyou A, <strong>Hicks SC</strong>, Salzberg SL, Pertea M. (2023). <a href="https://doi.org/10.1101/2023.07.10.548289">Detecting differential transcript usage in complex diseases with SPIT</a>. <em>bioRxiv</em>.</li>
</ol>
</section>
</section>
<section id="genomics-data-from-single-cells" class="level1">
<h1>Genomics data from single cells</h1>
<p>Single-cell RNA-Sequencing (scRNA-seq) data has become the most widely used high-throughput method for transcription profiling of individual cells. This technology has created an unprecedented opportunity to investigate important biological questions that can only be answered at the single-cell level. However, this technology also brings new statistical and computational challenges (1, 2).</p>
<ol type="1">
<li><p>Amezquita RA, Lun ATL, Carey VJ, Carpp LN, Geistlinger L, Marini F, Rue-Albrecht K, Risso D, Soneson C, Waldron L, Pages H, Smith M, Huber W, Morgan M, Gottardo R, <strong>Hicks SC</strong>. (2020). <a href="https://doi.org/10.1038/s41592-019-0654-x">Orchestrating Single-Cell Analysis with Bioconductor</a>. <em>Nature Methods</em>.</p></li>
<li><p>Lähnemann D, Koester J, Szczurek E, McCarthy D, <strong>Hicks S</strong>, Robinson MD, Vallejos C, Beerenwinkel N, et al.&nbsp;(2020). <a href="https://doi.org/10.1186/s13059-020-1926-6">Eleven grand challenges in single-cell data science</a>. <em>Genome Biology</em>.</p></li>
</ol>
<section id="methods-to-identify-and-remove-systematic-errors-and-biases" class="level2">
<h2 class="anchored" data-anchor-id="methods-to-identify-and-remove-systematic-errors-and-biases">Methods to identify and remove systematic errors and biases</h2>
<p>I was one of the first to demonstrate that systematic errors can explain a substantial percentage of observed cell-to-cell expression variability scRNA-seq, which in turn can lead to false discoveries, for example, when using unsupervised learning methods (1). To address this, we developed a fast, scalable statistical framework for feature selection and dimensionality reduction using generalized principal component analysis (GLM-PCA) for scRNA-seq data, which permits the identification of low-dimensional representations of cells measured with unique molecular identifiers (UMI) count data using a multinomial model (2). More recently, we developed a probabilistic model to identify low-quality cells in scRNA-seq data (3). Finally, we have evaluated probability distributions to accurately model single-nucleus RNA-sequencing (snRNA-seq) data and showed how snRNA-seq has a gene length bias present in both exonic and intronic reads (4).</p>
<ol type="1">
<li><strong>Hicks SC</strong>, Townes FW, Teng M, Irizarry RA (2018). <a href="https://academic.oup.com/biostatistics/article/19/4/562/4599254">Missing data and technical variability in single-cell RNA-sequencing experiments</a>. <em>Biostatistics</em>.</li>
<li>Townes FW, <strong>Hicks SC</strong>, Aryee MJ, Irizarry RA (2019). <a href="https://doi.org/10.1186/s13059-019-1861-6">Feature Selection and Dimension Reduction for Single Cell RNA-Seq based on a Multinomial Model</a>. <em>Genome Biology</em>.</li>
<li><strong>Hippen AA</strong>, Falco MM, <strong>Weber LM</strong>, Erkan EP, Zhang K, Dohert JA, Vähärautio A, Greene CS, <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1371/journal.pcbi.1009290">miQC: An adaptive probabilistic framework for quality control of single-cell RNA-sequencing data</a>. <em>PLOS Computational Biology</em>.</li>
<li><strong>Kuo A</strong>, Hansen KD, <strong>Hicks SC</strong>. (2023). <a href="https://doi.org/10.1093/biostatistics/kxad010">Quantification and statistical modeling of Chromium-based single-nucleus RNA-sequencing data</a>. <em>Biostatistcs</em>.</li>
</ol>
</section>
<section id="methods-for-distance-metrics-and-unsupervised-analyses" class="level2">
<h2 class="anchored" data-anchor-id="methods-for-distance-metrics-and-unsupervised-analyses">Methods for distance metrics and unsupervised analyses</h2>
<p>Computational methods and open-source software to store, access, and analyze single-cell data are essential. Most importantly, they need to be fast, scalable, and memory efficient. For example, the <em>k</em>-means algorithm is a classic algorithm used in the analysis of scRNA-seq data. However, with increasing sizes of single-cell data, new methods are needed that are fast, scalable and memory-efficient. To address this, we implemented the mini-batch optimization for <em>k</em>-means clustering proposed in <a href="https://www.eecs.tufts.edu/~dsculley/papers/fastkmeans.pdf">Sculley (2010)</a> for large single cell sequencing data (1), including fast and memory-efficient k-means++ center finding with distance measures designed for count data and probability distributions (2). The mini-batch <em>k</em>-means algorithm can be run with data stored in memory or on disk (e.g.&nbsp;HDF5 file format). In addition, we have developed fast and scalable discordance metrics to use as a performance metric to evaluate the predicted labels for a general dissimilarity measure (e.g.&nbsp;Euclidean distance) (3). Finally, we have also implemented scalable data infrastructure to store and access single-cell data that can be explored with hierarchical structures (e.g.&nbsp;cell types) (4).</p>
<ol type="1">
<li><strong>Hicks SC</strong>, Liu R, Ni Y, Purdom E, Risso D. (2020). <a href="https://doi.org/10.1371/journal.pcbi.1008625">mbkmeans: fast clustering for single cell data using mini-batch <em>k</em>-means</a>. <em>PLOS Computational Biology</em>.</li>
<li>Baker DN, <strong>Dyjack N</strong>, Braverman V, <strong>Hicks SC</strong>, Langmead B. (2021). <a href="https://doi.org/10.1145/3459930.3469523">minicore: Fast scRNA-seq clustering with various distance measures</a>. <em>ACM Conference on Bioinformatics, Computational Biology and Biomedicine</em>.</li>
<li><strong>Dyjack N</strong>, Baker DN, Braverman V, Langmead B, <strong>Hicks SC</strong>. (2022). <a href="https://doi.org/10.1093/biostatistics/kxac035">A scalable and unbiased discordance metric with <em>H+</em></a>. <em>Biostatistics</em>.</li>
<li>Huang R, Soneson C, Ernst FGM, Rue-Albrecht KC, Guangchuang Y, <strong>Hicks SC</strong>, Robinson MD. (2020). <a href="https://doi.org/10.12688/f1000research.26669.1">TreeSummarizedExperiment: a S4 class for data with hierarchical structure</a>. <em>F1000Research</em>.</li>
</ol>
</section>
<section id="methods-to-identify-differences-between-groups-1" class="level2">
<h2 class="anchored" data-anchor-id="methods-to-identify-differences-between-groups-1">Methods to identify differences between groups</h2>
<p>Another important question is how to identify differences between biological groups or conditions in single-cell data. For example, these count-based models, such as Poisson or negative binomial, can be generalized using the Tweedie family of distributions, which we used to develop a statistical method to identify differentially expressed genes (1). Similarly, we have developed methods for differential pseudotime analysis (gene expression or cell abundance) with multiple samples per group using B-splines (2).</p>
<ol type="1">
<li>Mallick H, Chatterjee Su., Chowdhury S, Chatterjee Sa., Rahnavard A, <strong>Hicks SC</strong>. (2021). <a href="http://doi.org/10.1002/sim.9430">Differential expression of single-cell RNA-seq data using Tweedie models</a>. <em>Statistics in Medicine</em>.</li>
<li><strong>Hou W</strong>, Ji Z, Chen Z, Wherry EJ, <strong>Hicks SC</strong>, Ji H. (2021). <a href="https://doi.org/10.1038/s41467-023-42841-y">A statistical framework for differential pseudotime analysis with multiple single-cell RNA-seq samples</a>. <em>Nature Communications</em>.</li>
</ol>
</section>
<section id="single-cell-applications" class="level2">
<h2 class="anchored" data-anchor-id="single-cell-applications">Single cell applications</h2>
<section id="high-grade-serous-ovarian-cancer" class="level3">
<h3 class="anchored" data-anchor-id="high-grade-serous-ovarian-cancer">High-grade serous ovarian cancer</h3>
<p>The goal of this project is to identify the biological basis of subtypes of high-grade serous ovarian cancers (HGSOC) using bulk and single-cell gene expression data. This is highly relevant to public health because HGSOC is a particularly deadly cancer that is often only identified at late stage and treatment options are limited. The long-term impact of this project will be a key step towards developing targeted treatments for HGSOCs. Towards this goal, we demonstrated that genetic demultiplexing from single-cell cancer samples can be used for better experimental design and increase cost savings (1). Most recently, we performed a systematic benchmark of algorithms for bulk deconvolution for HGSOC tumors (2) and showed how applyings these algorithms in the context of high-grade serous ovarian cancer reveals compositional differences in subtypes.</p>
<ol type="1">
<li><strong>Weber LM</strong>, Hippen AA, Hickey PF, Berrett KC, Gertz J, Doherty JA, Greene CS, <strong>Hicks SC</strong>. (2020). <a href="https://doi.org/10.1093/gigascience/giab062">Genetic demultiplexing of pooled single-cell RNA-sequencing samples in cancer facilitates effective experimental design</a>. <em>GigaScience</em>.</li>
<li>Hippen AA, Omran DK, <strong>Weber LM</strong>, Jung E, Drapkin R, Doherty JA, <strong>Hicks SC</strong>, Greene CS. (2023). <a href="https://doi.org/10.1186/s13059-023-03077-7">Performance of computational algorithms to deconvolve heterogeneous bulk tumor tissue depends on experimental factors</a>. <em>Genome Biology</em>.</li>
<li>Hippen AA, Davidson NR, Barnard ME, <strong>Weber LM</strong>, Gertz J, Doherty JA, <strong>Hicks SC</strong>, Greene CS. (2023). <a href="https://doi.org/10.1101/2023.06.14.544991">Deconvolution reveals compositional differences in high-grade serous ovarian cancer subtypes</a>. <em>bioRxiv</em>.</li>
</ol>
</section>
<section id="neuroscience" class="level3">
<h3 class="anchored" data-anchor-id="neuroscience">Neuroscience</h3>
<p>Single-nucleus RNA-sequencing (snRNA-seq) has become the preferred experimental technology, compared to scRNA-seq, to profile gene expression in frozen cells or cells that are hard to dissociate, such as brain tissue. Previous studies have shown that snRNA-seq offers substantial advantages over scRNA-seq, including reduced dissociation bias and the ability to capture rare cell types in these tissues including pan regions of the brain (1). For example, we recently used snRNA-seq to explore activity-regulated gene expression across cell types of the mouse hippocampus (2).</p>
<ol type="1">
<li>Tran MN, Maynard KR, Spangler A, Collado-Torres L, Sadashivaiah V, Tippani M, Barry BK, Hancock DB, <strong>Hicks SC</strong>, Kleinman JE, Hyde TM, Martinowich K, Jaffe A. (2020). <a href="https://doi.org/10.1016/j.neuron.2021.09.001">Single-nucleus transcriptome analysis reveals cell type-specific molecular signatures across reward circuitry in the human brain</a>. <em>Neuron</em>.</li>
<li><strong>Nelson ED</strong>, Maynard KR, Nicholas KR, Tran MN, Divecha HR, Collado-Torres L, <strong>Hicks SC</strong>, Martinowich K. (2022). <a href="https://doi.org/10.1002/hipo.23548">Activity-regulated gene expression across cell types of the mouse hippocampus</a>. <em>Hippocampus</em>.</li>
</ol>
</section>
</section>
</section>
<section id="genomics-data-with-spatial-resolution" class="level1">
<h1>Genomics data with spatial resolution</h1>
<p>Spatially-resolved transcriptomics (SRT), the <a href="https://doi.org/10.1038/s41592-020-01042-x">Nature Methods 2020 Method of the Year</a>, is poised to transform our understanding of how the spatial organization and communication of cells in complex tissues. These technologies have created an unprecedented opportunity to investigate important biological questions that can only be answered in a spatial context. However, these technologies also brings new statistical, computational and methodological challenges.</p>
<section id="data-storage-and-data-visualization" class="level2">
<h2 class="anchored" data-anchor-id="data-storage-and-data-visualization">Data storage and data visualization</h2>
<p>To address some of these challenges, we have developed standardized data infrastructure to storage SRT data within the Bioconductor framework (1) along with tools to statistially visualize multi-dimensional data in a 2D space, including embedding visualizations or spatial visualizations (2). Other tools include interactively visualizing SRT data (3-4). Most recently, we developed a performant web-based interactive visualization tool for spatially-resolved transcriptomics experiments called <a href="https://samuibrowser.com">Samui Browser</a>, which leverages GeoTIFF file formats and <a href="https://openlayers.org">OpenLayers</a> for geomaps (5).</p>
<ol type="1">
<li>Righelli D, <strong>Weber LM</strong>, Crowell HL, Pardo B, Collado-Torres L, Ghazanfar S, Lun ATL, <strong>Hicks SC</strong>, Risso D (2021). <a href="https://doi.org/10.1093/bioinformatics/btac299">SpatialExperiment: infrastructure for spatially resolved transcriptomics data in R using Bioconductor</a>. <em>Bioinformatics</em>.</li>
<li><strong>Guo B</strong>, Hicks SC (2023). <a href="https://doi.org/10.1101/2023.03.18.533302">escheR: Unified multi-dimensional visualizations with Gestalt principles</a>. <em>bioRxiv</em>.</li>
<li>Pardo B, Spangler A, <strong>Weber LM</strong>, <strong>Hicks SC</strong>, Jaffe AE, Martinowich K, Maynard KR, Collado-Torres L. <a href="https://doi.org/10.1186/s12864-022-08601-w">spatialLIBD: an R/Bioconductor package to visualize spatially-resolved transcriptomics data</a>. <em>BMC Genomics</em>.</li>
<li>Tippani M, Divecha HR, Catallini JL, <strong>Weber LM</strong>, Spangler A, Jaffe AE, <strong>Hicks SC</strong>, Martinowich K, Collado-Torres L, Page SC, Maynard KR. (2021). <a href="https://doi.org/10.1101/2021.08.04.452489">VistoSeg: processing utilities for high-resolution Visium/Visium-IF images for spatial transcriptomics data</a>. Accepted in <em>Biological Imaging</em>.</li>
<li><strong>Sriworarat C</strong>, Nguyen AB, Eagles NJ, Collado-Torres L, Martinowich K, Maynard KR, <strong>Hicks SC</strong>. (2023). <a href="https://doi.org/10.1017/S2633903X2300017X">Performant web-based interactive visualization tool for spatially-resolved transcriptomics experiments</a>. <em>Biological Imaging</em>.</li>
</ol>
</section>
<section id="methods-to-identify-spatially-variable-genes" class="level2">
<h2 class="anchored" data-anchor-id="methods-to-identify-spatially-variable-genes">Methods to identify spatially variable genes</h2>
<p>Feature selection to identify spatially variable genes (SVGs) is a key step during analyses of SRT data. We developed a scalable approach to identify SVGs based on nearest-neighbor Gaussian processes (GPs), which uses gene-specific estimates of length scale parameters within GP models, and scales linearly with the number of spatial locations (1).</p>
<ol type="1">
<li><strong>Weber LM</strong>, Saha A, Datta A, Hansen KD, <strong>Hicks SC</strong>. (2023). <a href="https://doi.org/10.1038/s41467-023-39748-z">nnSVG: scalable identification of spatially variable genes using nearest-neighbor Gaussian processes</a>. <em>Nature Communications</em>.</li>
</ol>
</section>
<section id="spatial-atlases" class="level2">
<h2 class="anchored" data-anchor-id="spatial-atlases">Spatial atlases</h2>
<section id="neuroscience-1" class="level3">
<h3 class="anchored" data-anchor-id="neuroscience-1">Neuroscience</h3>
<p>We have created transcriptome-wide spatial atlases using the 10x Genomics Visium platform using postmorteum human brain regions, including dorsolateral prefrontal cortex (1,2), inferior temporal cortex (3), and locus coeruleus (4). This is highly relevant to public health because these technologies will provide insights into topographical and pathological changes in gene expression for example in the aging human brain or in patients affected by psychiatric diseases.</p>
<ol type="1">
<li>Maynard KR, Collado-Torres L, <strong>Weber LM</strong>, Uytingco C, Barry BK, Williams SR, II JLC, Tran MN, Besich Z, Tippani M, Chew J, Yin Y, Kleinman JE, Hyde TM, Rao N, <strong>Hicks SC</strong>, Martinowich K, Jaffe AE (2021). <a href="https://doi.org/10.1038/s41593-020-00787-0">Transcriptome-scale spatial gene expression in the human dorsolateral prefrontal cortex</a>. <em>Nature Neuroscience</em>.</li>
<li>Huuki-Myers LA, Spangler A, Eagles NJ, Montgomery KD, Kwon SH, <strong>Guo B</strong>, Grant-Peters M, Divecha HR, Tippani M, <strong>Sriworarat C</strong>, Nguyen AB, <strong>Ravichandran P</strong>, Tran MT, Seyedian A, PsychENCODE Consortium, Hyde TM, Kleinman JE, Battle A, Page SC, Ryten M, <strong>Hicks SC</strong>, Martinowich K, Collado-Torres L, Maynard KR. (2023). <a href="https://doi.org/10.1101/2023.02.15.528722">Integrated single cell and unsupervised spatial transcriptomic analysis defines molecular anatomy of the human dorsolateral prefrontal cortex</a>. <em>bioRxiv</em>.</li>
<li>Kwon SH, <strong>Parthiban S</strong>, Tippani M, Divecha HR, Eagles NJ, Lobana JS, Williams SR, Mark M, Bharadwaj RA, Kleinman JE, Hyde TM, Page SC, <strong>Hicks SC</strong>, Martinowich K, Maynard KR, Collado-Torres L. (2023). <a href="https://doi.org/10.1089/genbio.2023.0019">Influence of Alzheimer’s disease related neuropathology on local microenvironment gene expression in the human inferior temporal cortex</a>. <em>GEN Biotechnology</em>.</li>
<li><strong>Weber LM</strong>, Divecha HR, Tran MN, Kwon SH, Spangler A, Montgomery KD, Tippani M, Bharadwaj R, Kleinman JE, Page SC, Hyde TM, Collado-Torres L, Maynard KR, Martinowich K, <strong>Hicks SC</strong>. 2023. <a href="https://doi.org/10.7554/eLife.84628.1">The gene expression landscape of the human locus coeruleus revealed by single-nucleus and spatially-resolved transcriptomics</a>. <em>eLife</em>.</li>
</ol>
</section>
</section>
</section>
<section id="benchmarking-open-source-software-for-genomics" class="level1">
<h1>Benchmarking open-source software for genomics</h1>
<p>In addition to developing statistical methods, I am strongly invested in implementing methods in open-source software for the analysis of genomics data leading to improved reproducibility and understanding of biological data. Bioconductor is an open-source, open-development software project in the R programming language for the analysis and comprehension of high-throughput genomics and molecular biology data. As one of the leaders of the Bioconductor project, I have not only contributed software for the analysis of single-cell RNA-sequencing (scRNA-seq) data, but also I have led a team of investigators to address a growing need of how to approach single-cell genomic data in a comprehensible way published in <em>Nature Methods</em> (1), which included creating a freely available, <a href="http://bioconductor.org/books/release/OSCA/"><strong>42-chapter online book</strong></a> of single-cell methods and tools for prospective users and was highlighted in a <em>Nature Biotechnology</em> <a href="https://www.nature.com/articles/s41587-020-0449-8"><strong>editorial</strong></a> in 2020. More recently, I have contributed to the <a href="https://github.com/tidyomics">tidyomics ecosystem</a> bridging the tidyverse and omics data analysis in the R programming language (2).</p>
<ol type="1">
<li>Amezquita RA, Lun ATL, Carey VJ, Carpp LN, Geistlinger L, Marini F, Rue-Albrecht K, Risso D, Soneson C, Waldron L, Pages H, Smith M, Huber W, Morgan M, Gottardo R, <strong>Hicks SC</strong>. (2020). <a href="https://doi.org/10.1038/s41592-019-0654-x">Orchestrating Single-Cell Analysis with Bioconductor</a>. <em>Nature Methods</em>.</li>
<li>Hutchison WJ, Keyes TJ, The Tidyomics Consortium, Crowell LH, Soneson C, Yuan V, Nahid AA, Mu W, Park J, Davis ES, Tang M, Axisa PP, Kitt W, Poon CL, Kosmac M, Serizay J, Sato N, Gottardo R, Morgan M, Lee S, Lawrence M, <strong>Hicks SC</strong>, Nolan GP, Davis KL, Papenfuss AT, Love M, Mangiola S. (2023). <a href="https://doi.org/10.1101/2023.09.10.557072">The tidyomics ecosystem: Enhancing omic data analyses</a>. <em>bioRxiv</em>.</li>
</ol>
<p>In addition, I evaluate open-source software packages for their accuracy, usability and reproducibility. For example, in high-throughput studies, hundreds to millions of hypotheses are typically tested. Statistical methods that control the false discovery rate (FDR) have emerged as popular and powerful tools for error rate control. While classic FDR methods use only <em>p</em>-values as input, more modern FDR methods have been shown to increase power by incorporating complementary information as “informative covariates” to prioritize, weight, and group hypotheses. To address this, we investigated the accuracy, applicability, and ease of use of two classic and six modern FDR-controlling methods by performing a systematic benchmark comparison using simulation studies as well as six case studies in computational biology (2).</p>
<ol start="2" type="1">
<li>Korthauer K, Kimes PK, Duvallet C, Reyes A, Subramanian A, Teng M, Shukla C, Alm EJ, <strong>Hicks SC</strong> (2019). <a href="https://doi.org/10.1186/s13059-019-1716-1">A practical guide to methods controlling false discoveries in computational biology</a>. <em>Genome Biology</em>.</li>
</ol>
<p>Most recently, we performed a benchmark comparison of 18 scRNA-seq imputation methods across multiple experimental protocols and datasets (3).</p>
<ol start="3" type="1">
<li><strong>Hou W</strong>, Ji J, Ji H, <strong>Hicks SC</strong> (2020). <a href="https://doi.org/10.1186/s13059-020-02132-x">A systematic evaluation of single-cell RNA-sequencing imputation methods</a>. <em>Genome Biology</em>.</li>
</ol>
</section>
<section id="data-science-education" class="level1">
<h1>Data science education</h1>
<p>An increase in demand for statistics and data science education has led to changes in curriculum, specifically an increase in computing. While this has led to more applied courses, students still struggle with effectively deriving knowledge from data and solving real-world problems. In 1999, Deborah Nolan and Terry Speed argued the solution was to teach courses through in-depth case studies derived from interesting scientific questions with nontrivial solutions that leave room for different analyses of the data. This innovative framework teaches the student to make important connections between the scientific question, data and statistical concepts that only come from hands-on experience analyzing data (1-2).</p>
<ol type="1">
<li><strong>Hicks SC</strong>, Irizarry RA (2018). <a href="https://www.tandfonline.com/doi/abs/10.1080/00031305.2017.1356747?journalCode=utas20">A Guide to Teaching Data Science</a>. <em>The American Statistician</em>.</li>
<li><strong>Hicks SC</strong> (2017). <a href="https://www.tandfonline.com/doi/abs/10.1080/10618600.2017.1385472">Greater Data Science Ahead</a>. <em>Journal of Computational Graphical Statistics</em>.</li>
</ol>
<section id="open-case-studies" class="level2">
<h2 class="anchored" data-anchor-id="open-case-studies">Open Case Studies</h2>
<p>To address this, I built the <a href="https://www.opencasestudies.org">Open Case Studies</a> (OCS) (1) data science educational resource of case studies that educators can use in the classroom to teach students how to effectively derive knowledge from data. This project was selected as a High-Impact Project in 2019-2020 by the Bloomberg American Health Initiative and Bloomberg Philanthropies (2). A list of available cases studies are listed in the <a href="../teaching/index.html">teaching</a> section.</p>
<ol type="1">
<li>Wright C, Meng Q, Breshock MR, Atta L, Taub MA, Jager LR, Muschelli J, <strong>Hicks SC</strong> (2023). <a href="https://arxiv.org/abs/2301.05298">Open Case Studies: Statistics and Data Science Education through Real-World Applications</a>. <em>arXiv</em>.</li>
<li><a href="https://americanhealth.jhu.edu/open-case-studies" class="uri">https://americanhealth.jhu.edu/open-case-studies</a></li>
</ol>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/DgzBSOY5Yc8" frameborder="0" allowfullscreen="">
</iframe>
</p>
</section>
<section id="analytic-design-theory" class="level2">
<h2 class="anchored" data-anchor-id="analytic-design-theory">Analytic Design Theory</h2>
<p>The data science revolution has led to an increased interest in the practice of data analysis. While much has been written about <em>statistical thinking</em>, a complementary form of thinking that appears in the practice of data analysis is <em>design thinking</em> – the problem-solving process to understand the people for whom a product is being designed. There can be significant or subtle differences in how a data analyst (or producer of a data analysis) constructs, creates, or designs a data analysis, including differences in the choice of methods, tooling, and workflow. These choices can affect the data analysis products themselves and the experience of the consumer of the data analysis. Therefore, the role of a producer can be thought of as designing the data analysis with a set of design principles (1,2). Our work leads to several insights: it suggests a formal mechanism to describe and evaluate (3) data analyses based on the design principles for data analysis, and a framework for diagnosing data analytic problems in the classroom (4).</p>
<ol type="1">
<li>D’Agostino McGowan L, Peng RD, <strong>Hicks SC</strong>. (2022). <a href="https://doi.org/10.1080/10618600.2022.2104290">Design Principles for Data Analysis</a>. <em>Journal of Computational and Graphical Statistics</em>.</li>
<li><strong>Hicks SC</strong>, Peng RD. (2019). <a href="https://arxiv.org/abs/1903.07639">Elements and Principles for Characterizing Variation between Data Analyses</a>. <em>arXiv</em>.</li>
<li><strong>Hicks SC</strong>, Peng RD. (2019). <a href="https://arxiv.org/abs/1904.11907">Evaluating the Success of a Data Analysis</a>. <em>arXiv</em>.</li>
<li>Peng RD, Chen A, Bridgeford E, Leek JT, and <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1080/26939169.2021.1971586">Diagnosing Data Analytic Problems in the Classroom</a>. <em>Journal of Statistics and Data Science Education</em>.</li>
</ol>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>