<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Publications – Stephanie C. Hicks, PhD</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-d1029c8c0624b678ec1882d5dbfbed62.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Stephanie C. Hicks, PhD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./projects.html"> 
<span class="menu-text">Research themes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./people.html"> 
<span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./publications.html" aria-current="page"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./software.html"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./teaching.html"> 
<span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./talks.html"> 
<span class="menu-text">Talks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./join.html"> 
<span class="menu-text">Join us</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/stephaniehicks"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/stephaniehicks"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#preprints" id="toc-preprints" class="nav-link active" data-scroll-target="#preprints">Preprints</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section">2025</a></li>
  <li><a href="#section-1" id="toc-section-1" class="nav-link" data-scroll-target="#section-1">2024</a></li>
  <li><a href="#section-2" id="toc-section-2" class="nav-link" data-scroll-target="#section-2">2023</a></li>
  <li><a href="#section-3" id="toc-section-3" class="nav-link" data-scroll-target="#section-3">2022</a></li>
  <li><a href="#section-4" id="toc-section-4" class="nav-link" data-scroll-target="#section-4">2021</a></li>
  <li><a href="#earlier-years" id="toc-earlier-years" class="nav-link" data-scroll-target="#earlier-years">Earlier years</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Publications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>The following are a selected set of publications. To see a list with all of papers from Stephanie Hicks, you can visit <a href="https://scholar.google.com/citations?user=4T4qnL4AAAAJ&amp;hl=en">Google Scholar</a> or <a href="https://orcid.org/0000-0002-7858-0231">ORCiD</a>.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>The <span class="math inline">\(^*\)</span> indicates a mentored student or post-doctoral fellow.</li>
<li>The <span class="math inline">\(\dagger\)</span> and <span class="math inline">\(\ddagger\)</span> indicate co-first and co-last authors, respectively.</li>
</ul>
</div>
</div>
<section id="preprints" class="level2">
<h2 class="anchored" data-anchor-id="preprints">Preprints</h2>
<p>Mulvey B, Wang Y, Divecha HR, Bach SV, Montgomery KD, Cinquemani S, Chandra A, Du Y, Miller RA, Kleinman JE, Page SC, Hyde TM, Martinowich K, <strong>Hicks SC</strong>, Hansen KD<span class="math inline">\(^{\ddagger}\)</span>, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>. (2024). <a href="https://doi.org/10.1101/2024.12.07.627362">Spatially-resolved molecular sex differences at single cell resolution in the adult human hypothalamus</a> <em>bioRxiv</em>.</p>
<p>Wang, A, <strong>Hicks SC</strong>, Geman D, Younes L. (2024). <a href="https://doi.org/10.1101/2024.10.30.621151">GeneCover: A Combinatorial Approach for Label-free Marker Gene Selection</a>. <em>bioRxiv</em>. (Accepted for and presented at <a href="https://recomb.org/recomb2025/">RECOMB 2025</a>).</p>
<p>Totty MS<span class="math inline">\(^{*,\dagger}\)</span>, Cervera-Juanes R<span class="math inline">\(^{\dagger}\)</span>, Bach SV, Ameur LB, Valentine MR, Simons E, Romac MD, Trinh H, Henderson K, Rosario ID, Tippani M, Miller RA, Kleinman JE, Page SC, Saunders A, Hyde TM, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Costa VD<span class="math inline">\(^{\ddagger}\)</span>. (2024). <a href="https://doi.org/10.1101/2024.10.18.618721">Transcriptomic diversity of amygdalar subdivisions across humans and nonhuman primates</a>. <em>bioRxiv</em>.</p>
<p>Zhou H<span class="math inline">\(^{*}\)</span>, Panwar P, Guo B<span class="math inline">\(^{*}\)</span>, Hallinan C<span class="math inline">\(^{*}\)</span>, Ghazanfar S<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2024). <a href="https://doi.org/10.1101/2024.10.08.615307">Spatial mutual nearest neighbors for spatial transcriptomics data</a>. <em>bioRxiv</em>.</p>
<p>Peng RD, <strong>Hicks SC</strong>. (2024). <a href="https://arxiv.org/abs/2309.08494">Modeling Data Analytic Iteration With Probabilistic Outcome Sets</a>. <em>arXiv</em>.</p>
<p>Hippen AA, Davidson NR, Barnard ME, Weber LM<span class="math inline">\(^{*}\)</span>, Gertz J, Doherty JA, <strong>Hicks SC</strong>, Greene CS. (2023). <a href="https://doi.org/10.1101/2023.06.14.544991">Deconvolution reveals compositional differences in high-grade serous ovarian cancer subtypes</a>. <em>bioRxiv</em>.</p>
<p>D’Agostino McGowan L, Peng RD, <strong>Hicks SC</strong>. (2023). <a href="https://arxiv.org/abs/2312.07616">Quantifying the Alignment of a Data Analysis between Analyst and Audience</a>. <em>arXiv</em>.</p>
<p><strong>Hicks SC</strong>, Peng RD. (2019). <a href="https://arxiv.org/abs/1903.07639">Elements and Principles for Characterizing Variation between Data Analyses</a>. <em>arXiv</em>.</p>
</section>
<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section">2025</h2>
<p>Luecken MD, Gigante S, Burkhardt DB, Cannoodt R, Strobl DC, Markov NS, Zappia L, Palla G, Lewis W, Dimitrov D, Vinyard ME, Magruder DS, Andersson A, Dann E, Qin Q, Otto DJ, Klein M, Botvinnik OB, Deconinck L, Waldrant K, <strong>The Open Problems Jamboree Members</strong>, Bloom JM, Pisco AO, Saez-Rodriguez J, Wulsin D, Pinello L, Saeys Y, Theis FJ, Krishnaswamy S. (2025). <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11030530">Defining and benchmarking open problems in single-cell analysis</a>. Accepted in <em>Nature Biotechnology</em>.</p>
<p>Thompson JR<span class="math inline">\(^{*,\dagger}\)</span>, Nelson ED<span class="math inline">\(^{*,\dagger}\)</span>, Tippani M<span class="math inline">\(^{\dagger}\)</span>, Ramnauth AD, Divecha HR, Miller RA, Eagles NJ, Pattie EA, Kwon SH, Bach SV, Kaipa UM, Yao J<span class="math inline">\(^{*}\)</span>, Hou C<span class="math inline">\(^{*}\)</span>, Kleinman JE, Collado-Torres L, Han S, Maynard KR, Hyde TM, Martinowich K, Page SC<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2025). <a href="https://doi.org/10.1101/2024.04.26.590643">An integrated single-nucleus and spatial transcriptomics atlas reveals the molecular landscape of the human hippocampus</a>. Accepted in <em>Nature Neuroscience</em>.</p>
<p>Totty MS<span class="math inline">\(^{*}\)</span>, <strong>Hicks SC</strong>, Guo B<span class="math inline">\(^{*}\)</span>. (2025). <a href="https://doi.org/10.1038/s41592-025-02713-3">SpotSweeper: spatially-aware quality control for spatial transcriptomics</a>. <em>Nature Methods</em>.</p>
<p>Yao J<span class="math inline">\(^{*}\)</span>, Yu J, Caffo B, Page SC, Martinowich K, <strong>Hicks SC</strong>. (2025). <a href="https://doi.org/10.1101/gr.279380.124">Spatial domain detection using contrastive self-supervised learning for spatial multi-omics technologies</a>. <em>Genome Reserach</em>.</p>
<p>Popescu A, Ottaway C, Ford K, Patterson TW, Ingiosi AM, Medina E, <strong>Hicks SC</strong>, Singletary K, Peixoto L. (2025). <a href="https://journals.physiology.org/doi/abs/10.1152/physiolgenomics.00128.2024">Transcriptional Dynamics of Sleep Deprivation and Subsequent Recovery Sleep in the Male Mouse Cortex</a>. <em>Physiological Genomics</em>.</p>
<p>Maden SK<span class="math inline">\(^{*}\)</span>, Huuki-Myers LA, Kwon SH, Collado-Torres L, Maynard KR, <strong>Hicks SC</strong>. (2025). <a href="https://doi.org/10.1186/s12864-025-11508-x">lute: estimating the cell composition of heterogeneous tissue with varying cell sizes using gene expression</a>. <em>BMC Genomics</em>.</p>
<p>Ivich A, Davidson NR, Grieshober L, Li W<span class="math inline">\(^{*}\)</span>, <strong>Hicks SC</strong>, Doherty JA, Greene CS. (2025). <a href="https://doi.org/10.1101/2024.04.25.590992">Missing cell types in single-cell references impact deconvolution of bulk data but are detectable</a>. Accepted in <em>Genome Biology</em>.</p>
<p>Shah K<span class="math inline">\(^{*}\)</span>, Guo B<span class="math inline">\(^{*}\)</span>, <strong>Hicks SC</strong>. (2025). <a href="https://doi.org/10.1093/biostatistics/kxaf012">Addressing the mean-variance relationship in spatially resolved transcriptomics data with <em>spoon</em></a>. <em>Biostatistics</em>.</p>
<p>Huuki-Myers LA, Montgomery KD, Kwon SH, Cinquemani S, Maden SK<span class="math inline">\(^{*}\)</span>, Eagles NJ, Kleinman JE, Hyde TM, <strong>Hicks SC</strong>, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>, Collado-Torres L<span class="math inline">\(^{\ddagger}\)</span>. (2025). <a href="https://doi.org/10.1186/s13059-025-03552-3">Benchmark of cellular deconvolution methods using a multi-assay reference dataset from postmortem human prefrontal cortex</a>. <em>Genome Biology</em>.</p>
<p>Greene CS, Gignoux CR, Subirana-Granés M, Pividori M, <strong>Hicks SC</strong>, Ackert-Bicknell CL. (2025). <a href="https://www.sciencedirect.com/science/article/pii/S2352187225000166">Can AI reveal the next generation of high-impact bone genomics targets?</a>. <em>Bone Reports</em>.</p>
<p>Hou C<span class="math inline">\(^{*}\)</span>, Ghazanfar S, Marini F, Morgan M, <strong>Hicks SC</strong>. (2025). <a href="https://doi.org/10.1093/bioadv/vbaf048">HuBMAPR: an R Client for the HuBMAP Data Portal</a>. <em>Bioinformatics Advances</em>.</p>
<p>Ramnauth AD<span class="math inline">\(^{\dagger}\)</span>, Tippani M, Divecha HR, Papariello AR, Miller RA, Nelson ED<span class="math inline">\(^{*}\)</span>, Thompson JR<span class="math inline">\(^{*}\)</span>, Pattie EA, Kleinman JE, Maynard KR, Collado-Torres L, Hyde TM, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Page SC<span class="math inline">\(^{\ddagger}\)</span>. (2025). <a href="https://doi.org/10.1016/j.celrep.2025.115300">Spatially-resolved transcriptomics of human dentate gyrus across postnatal lifespan reveals heterogeneity in markers for proliferation, extracellular matrix, and neuroinflammation</a>. <em>Cell Reports</em>.</p>
<p>Guo B<span class="math inline">\(^{*,\dagger}\)</span>, Ling W<span class="math inline">\(^{\dagger}\)</span>, Kwon SH, Panwar P, Ghazanfar S<span class="math inline">\(^{\ddagger}\)</span>, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2025). <a href="https://doi.org/10.1002/smtd.202401194">Integrating spatially-resolved transcriptomics data across tissues and individuals: challenges and opportunities</a>. <em>Small Methods</em>.</p>
<p>Phillips III RA, Bach SV, Du Y, Miller RA, Kleinman JE, Hyde TM, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Page SC<span class="math inline">\(^{\ddagger}\)</span>, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>. (2025). <a href="https://doi.org/10.1016/j.isci.2025.111820">Transcriptomic characterization of human lateral septum neurons reveals conserved and divergent marker genes across species</a>. <em>iScience</em>.</p>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1">2024</h2>
<p>Wright C, Meng Q, Breshock MR, Atta L, Taub MA, Jager LR, Muschelli J, <strong>Hicks SC</strong> (2024). <a href="https://doi.org/10.1080/26939169.2024.2394541">Open Case Studies: Statistics and Data Science Education through Real-World Applications</a>. <em>Journal of Statistics and Data Science Education</em>.</p>
<p>Ford K, Zuin E, Righelli D, Medina E, Schoch H, Singletary K, Muheim C, Frank MG, <strong>Hicks SC</strong>, Risso D, Peixoto L. (2024). <a href="https://www.cell.com/iscience/fulltext/S2589-0042(24)01977-1">A global transcriptional atlas of the effect of sleep deprivation in the mouse frontal cortex.</a>. <em>iScience</em>.</p>
<p>Hutchison WJ, Keyes TJ, The Tidyomics Consortium, Crowell LH, Soneson C, Yuan V, Nahid AA, Mu W, Park J, Davis ES, Tang M, Axisa PP, Kitt W, Poon CL, Kosmac M, Serizay J, Sato N, Gottardo R, Morgan M, Lee S, Lawrence M, <strong>Hicks SC</strong>, Nolan GP, Davis KL, Papenfuss AT<span class="math inline">\(^{\ddagger}\)</span>, Love M<span class="math inline">\(^{\ddagger}\)</span>, Mangiola S<span class="math inline">\(^{\ddagger}\)</span>. (2024). <a href="https://doi.org/10.1038/s41592-024-02299-2">The tidyomics ecosystem: Enhancing omic data analyses</a>. <em>Nature Methods</em>.</p>
<p>Huuki-Myers LA, Spangler A, Eagles NJ, Montgomery KD, Kwon SH, Guo B<span class="math inline">\(^{*}\)</span>, Grant-Peters M, Divecha HR, Tippani M, Sriworarat C<span class="math inline">\(^{*}\)</span>, Nguyen AB, Ravichandran P<span class="math inline">\(^{*}\)</span>, Tran MT, Seyedian A, PsychENCODE Consortium, Hyde TM, Kleinman JE, Battle A, Page SC, Ryten M, <strong>Hicks SC</strong>, Martinowich K, Collado-Torres L<span class="math inline">\(^{\ddagger}\)</span>, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>. (2024). <a href="https://www.science.org/doi/10.1126/science.adh1938">A data-driven single-cell and spatial transcriptomic map of the human prefrontal cortex</a>. <em>Science</em>.</p>
<p>Ruzicka WB, Mohammadi S, Fullard JF, Davila-Velderrain J, Subburaju S, Tso DR, Hourihan M, Jiang S, Lee H-C, Bendl J, <strong>PsychENCODE Consortium</strong>, Voloudakis G, Haroutunian V, Hoffman GE, Roussos P, Kellis M. (2024). <a href="https://www.science.org/doi/10.1126/science.adg5136">Single-cell multi-cohort dissection of the schizophrenia transcriptome</a>. <em>Science</em>.</p>
<p>Weber LM<span class="math inline">\(^{*,\dagger}\)</span>, Divecha HR<span class="math inline">\(^{\dagger}\)</span>, Tran MN, Kwon SH, Spangler A, Montgomery KD, Tippani M, Bharadwaj R, Kleinman JE, Page SC, Hyde TM, Collado-Torres L, Maynard KR, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2024). <a href="https://elifesciences.org/articles/84628">The gene expression landscape of the human locus coeruleus revealed by single-nucleus and spatially-resolved transcriptomics</a>. <em>eLife</em>.</p>
<p>Erdogdu B, Varabyou A, <strong>Hicks SC</strong>, Salzberg SL, Pertea M. (2024). <a href="https://pubmed.ncbi.nlm.nih.gov/38508189/">Detecting differential transcript usage in complex diseases with SPIT</a>. <em>Cell Reports Methods</em>.</p>
<p>Kuo A<span class="math inline">\(^{*}\)</span>, Hansen KD, <strong>Hicks SC</strong>. (2024). <a href="https://doi.org/10.1093/biostatistics/kxad010">Quantification and statistical modeling of Chromium-based single-nucleus RNA-sequencing data</a>. <em>Biostatistcs</em>.</p>
</section>
<section id="section-2" class="level2">
<h2 class="anchored" data-anchor-id="section-2">2023</h2>
<p>Weber LM<span class="math inline">\(^{*}\)</span>, Saha A, Datta A, Hansen KD, <strong>Hicks SC</strong>. (2023). <a href="https://doi.org/10.1038/s41467-023-39748-z">nnSVG: scalable identification of spatially variable genes using nearest-neighbor Gaussian processes</a>. <em>Nature Communications</em>.</p>
<p>Guo B<span class="math inline">\(^{*}\)</span>, Huuki-Myers LA, Grant-Peters M, Collado-Torres L, <strong>Hicks SC</strong>. (2023). <a href="https://doi.org/10.1093/bioadv/vbad179">escheR: Unified multi-dimensional visualizations with Gestalt principles</a>. <em>Bioinformatics Advances</em>.</p>
<p>Sriworarat C<span class="math inline">\(^{*}\)</span>, Nguyen AB, Eagles NJ, Collado-Torres L, Martinowich K, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2023). <a href="https://doi.org/10.1017/S2633903X2300017X">Performant web-based interactive visualization tool for spatially-resolved transcriptomics experiments</a>. <em>Biological Imaging</em>.</p>
<p>Tippani M, Divecha HR, Catallini JL, Weber LM<span class="math inline">\(^{*}\)</span>, Spangler A, Jaffe AE, <strong>Hicks SC</strong>, Martinowich K, Collado-Torres L, Page SC<span class="math inline">\(^{\ddagger}\)</span>, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>. (2023). <a href="https://doi.org/10.1017/S2633903X23000235">VistoSeg: processing utilities for high-resolution Visium/Visium-IF images for spatial transcriptomics data</a>. <em>Biological Imaging</em>.</p>
<p>Kwon SH, Parthiban S<span class="math inline">\(^{*}\)</span>, Tippani M, Divecha HR, Eagles NJ, Lobana JS, Williams SR, Mark M, Bharadwaj RA, Kleinman JE, Hyde TM, Page SC, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>, Collado-Torres L<span class="math inline">\(^{\ddagger}\)</span>. (2023). <a href="https://doi.org/10.1089/genbio.2023.0019">Influence of Alzheimer’s disease related neuropathology on local microenvironment gene expression in the human inferior temporal cortex</a>. <em>GEN Biotechnology</em>.</p>
<p>Maden SK<span class="math inline">\(^{*}\)</span>, Kwon SH, Huuki-Myers LA, Collado-Torres L, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Maynard KR<span class="math inline">\(^{\ddagger}\)</span>. (2023). <a href="https://doi.org/10.1186/s13059-023-03123-4">Challenges and opportunities to computationally deconvolve heterogeneous tissue with varying cell sizes using single cell RNA-sequencing datasets</a>. <em>Genome Biology</em>.</p>
<p>Hou W<span class="math inline">\(^{*}\)</span>, Ji Z, Chen Z, Wherry EJ, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Ji H<span class="math inline">\(^{\ddagger}\)</span>. (2023). <a href="https://doi.org/10.1038/s41467-023-42841-y">A statistical framework for differential pseudotime analysis with multiple single-cell RNA-seq samples</a>. <em>Nature Communications</em>.</p>
<p>Hippen AA, Omran DK, Weber LM<span class="math inline">\(^{*}\)</span>, Jung E, Drapkin R, Doherty JA, <strong>Hicks SC</strong>, Greene CS. (2023). <a href="https://doi.org/10.1186/s13059-023-03077-7">Performance of computational algorithms to deconvolve heterogeneous bulk tumor tissue depends on experimental factors</a>. <em>Genome Biology</em>.</p>
<p>Nelson ED<span class="math inline">\(^{*}\)</span>, Maynard KR, Nicholas KR, Tran MN, Divecha HR, Collado-Torres L, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>. (2023). <a href="https://doi.org/10.1002/hipo.23548">Activity-regulated gene expression across cell types of the mouse hippocampus</a>. <em>Hippocampus</em>.</p>
<p>Dyjack N<span class="math inline">\(^{*}\)</span>, Baker DN, Braverman V, Langmead B, <strong>Hicks SC</strong>. (2023). <a href="https://pubmed.ncbi.nlm.nih.gov/36063544/">A scalable and unbiased discordance metric with H+</a>. <em>Biostatistics</em>.</p>
<p>Huuki-Myers LA, Montgomery KD, Kwon SH, Page SC, <strong>Hicks SC</strong>, Maynard KR, Collado-Torres L. (2023). <a href="https://doi.org/10.1186/s13059-023-03066-w">Data-driven Identification of Total RNA Expression Genes (TREGs) for Estimation of RNA Abundance in Heterogeneous Cell Types</a>. <em>Genome Biology</em>.</p>
</section>
<section id="section-3" class="level2">
<h2 class="anchored" data-anchor-id="section-3">2022</h2>
<p>Righelli D<span class="math inline">\(^{\dagger}\)</span>, Weber LM<span class="math inline">\(^{*,\dagger}\)</span>, Crowell HL<span class="math inline">\(^{\dagger}\)</span>, Pardo B, Collado-Torres L, Ghazanfar S, Lun ATL, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>, Risso D<span class="math inline">\(^{\ddagger}\)</span> (2022). <a href="https://doi.org/10.1093/bioinformatics/btac299">SpatialExperiment: infrastructure for spatially resolved transcriptomics data in R using Bioconductor</a>. <em>Bioinformatics</em>.</p>
<p>Wang Y, <strong>Hicks SC</strong>, Hansen KD (2022). <a href="https://doi.org/10.1371/journal.pcbi.1009954">Addressing the mean-correlation relationship in co-expression analysis</a>. <em>PLOS Computational Biology</em>.</p>
<p>D’Agostino McGowan L, Peng RD, <strong>Hicks SC</strong>. (2022). <a href="https://doi.org/10.1080/10618600.2022.2104290">Design Principles for Data Analysis</a>. <em>Journal of Computational and Graphical Statistics</em>.</p>
<p>Pardo B, Spangler A, Weber LM<span class="math inline">\(^{*}\)</span>, <strong>Hicks SC</strong>, Jaffe AE, Martinowich K, Maynard KR, Collado-Torres L. (2022). <a href="https://doi.org/10.1186/s12864-022-08601-w">spatialLIBD: an R/Bioconductor package to visualize spatially-resolved transcriptomics data</a>. <em>BMC Genomics</em>.</p>
<p>Mallick H, Chatterjee Su., Chowdhury S, Chatterjee Sa., Rahnavard A<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2022). <a href="http://doi.org/10.1002/sim.9430">Differential expression of single-cell RNA-seq data using Tweedie models</a>. <em>Statistics in Medicine</em>.</p>
</section>
<section id="section-4" class="level2">
<h2 class="anchored" data-anchor-id="section-4">2021</h2>
<p>Heil BJ, Hoffman MM, Markowetz F, Lee S-I, Greene CS<span class="math inline">\(^{\ddagger}\)</span>, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2021). <a href="https://doi.org/10.1038/s41592-021-01256-7">Reproducibility standards for machine learning in the life sciences</a>. <em>Nature Methods</em>.</p>
<p>Maynard KR<span class="math inline">\(^{\dagger}\)</span>, Collado-Torres L<span class="math inline">\(^{\dagger}\)</span>, Weber LM<span class="math inline">\(^{*}\)</span>, Uytingco C, Barry BK, Williams SR, II JLC, Tran MN, Besich Z, Tippani M, Chew J, Yin Y, Kleinman JE, Hyde TM, Rao N, <strong>Hicks SC</strong>, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, Jaffe AE<span class="math inline">\(^{\ddagger}\)</span> (2021). <a href="https://doi.org/10.1038/s41593-020-00787-0">Transcriptome-scale spatial gene expression in the human dorsolateral prefrontal cortex</a>. <em>Nature Neuroscience</em>.</p>
<p>Hippen AA<span class="math inline">\(^{*}\)</span>, Falco MM, Weber LM<span class="math inline">\(^{*}\)</span>, Erkan EP, Zhang K, Dohert JA, Vähärautio A<span class="math inline">\(^{\ddagger}\)</span>, Greene CS, <strong>Hicks SC</strong><span class="math inline">\(^{\ddagger}\)</span>. (2021). <a href="https://doi.org/10.1371/journal.pcbi.1009290">miQC: An adaptive probabilistic framework for quality control of single-cell RNA-sequencing data</a>. <em>PLOS Computational Biology</em>.</p>
<p>Baker DN, Dyjack N<span class="math inline">\(^{*}\)</span>, Braverman V, <strong>Hicks SC</strong>, Langmead B. (2021). <a href="https://doi.org/10.1145/3459930.3469523">minicore: Fast scRNA-seq clustering with various distance measures</a>. <em>ACM Conference on Bioinformatics, Computational Biology and Biomedicine</em>.</p>
<p>Weber LM<span class="math inline">\(^{*}\)</span>, Hippen AA, Hickey PF, Berrett KC, Gertz J, Doherty JA, Greene CS, <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1093/gigascience/giab062">Genetic demultiplexing of pooled single-cell RNA-sequencing samples in cancer facilitates effective experimental design</a>. <em>GigaScience</em>.</p>
<p>Tran MN<span class="math inline">\(^{\dagger}\)</span>, Maynard KR<span class="math inline">\(^{\dagger}\)</span>, Spangler A, Collado-Torres L, Sadashivaiah V, Tippani M, Barry BK, Hancock DB, <strong>Hicks SC</strong>, Kleinman JE, Hyde TM, Martinowich K<span class="math inline">\(^{\ddagger}\)</span>, Jaffe A<span class="math inline">\(^{\ddagger}\)</span>. (2021). <a href="https://doi.org/10.1016/j.neuron.2021.09.001">Single-nucleus transcriptome analysis reveals cell type-specific molecular signatures across reward circuitry in the human brain</a>. <em>Neuron</em>.</p>
<p>Peng RD, Chen A, Bridgeford E, Leek JT, and <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1080/26939169.2021.1971586">Diagnosing Data Analytic Problems in the Classroom</a>. <em>Journal of Statistics and Data Science Education</em>.</p>
<p>Peng RD, <strong>Hicks SC</strong>. (2021). <a href="https://doi.org/10.1146/annurev-publhealth-012420-105110">Reproducible Research: A Retrospective</a>. <em>Annual Review of Public Health</em>.</p>
<p><strong>Hicks SC</strong>, Liu R, Ni Y, Purdom E, Risso D. (2021). <a href="https://doi.org/10.1371/journal.pcbi.1008625">mbkmeans: fast clustering for single cell data using mini-batch <em>k</em>-means</a>. <em>PLOS Computational Biology</em>.</p>
</section>
<section id="earlier-years" class="level2">
<h2 class="anchored" data-anchor-id="earlier-years">Earlier years</h2>
<p>Amezquita RA, Lun ATL, Carey VJ, Carpp LN, Geistlinger L, Marini F, Rue-Albrecht K, Risso D, Soneson C, Waldron L, Pages H, Smith M, Huber W, Morgan M, Gottardo R, <strong>Hicks SC</strong>. (2020). <a href="https://doi.org/10.1038/s41592-019-0654-x">Orchestrating Single-Cell Analysis with Bioconductor</a>. <em>Nature Methods</em>.</p>
<p>Lahnemann D, Koester J, Szczurek E, McCarthy D, <strong>Hicks S</strong> <em>et al</em>. (2020). <a href="https://pubmed.ncbi.nlm.nih.gov/32033589/">Eleven grand challenges in single-cell data science</a>. <em>Genome Biology</em>.</p>
<p>Hou W<span class="math inline">\(^{*}\)</span>, Ji J, Ji H, <strong>Hicks SC</strong> (2020). <a href="https://doi.org/10.1186/s13059-020-02132-x">A systematic evaluation of single-cell RNA-sequencing imputation methods</a>. <em>Genome Biology</em>.</p>
<p>Huang R, Soneson C, Ernst FGM, Rue-Albrecht KC, Guangchuang Y, <strong>Hicks SC</strong>, Robinson MD. (2020). <a href="https://doi.org/10.12688/f1000research.26669.1">TreeSummarizedExperiment: a S4 class for data with hierarchical structure</a>. <em>F1000Research</em>.</p>
<p>Townes FW, <strong>Hicks SC</strong>, Aryee MJ, Irizarry RA (2019). <a href="https://doi.org/10.1186/s13059-019-1861-6">Feature Selection and Dimension Reduction for Single Cell RNA-Seq based on a Multinomial Model</a>. <em>Genome Biology</em>.</p>
<p><strong>Hicks SC</strong>, Irizarry RA. (2019). <a href="https://doi.org/10.1186/s13059-019-1827-8">methylCC: technology-independent estimation of cell type composition using differentially methylated regions</a>. <em>Genome Biology</em>.</p>
<p>Korthauer K<span class="math inline">\(^{\dagger}\)</span>, Kimes PK<span class="math inline">\(^{\dagger}\)</span>, Duvallet C, Reyes A, Subramanian A, Teng M, Shukla C, Alm EJ, <strong>Hicks SC</strong> (2019). <a href="https://doi.org/10.1186/s13059-019-1716-1">A practical guide to methods controlling false discoveries in computational biology</a>. <em>Genome Biology</em>.</p>
<p><strong>Hicks SC</strong>, Townes FW, Teng M, Irizarry RA (2018). <a href="https://academic.oup.com/biostatistics/article/19/4/562/4599254">Missing data and technical variability in single-cell RNA-sequencing experiments</a>. <em>Biostatistics</em>.</p>
<p><strong>Hicks SC</strong>, Okrah K, Paulson JN, Quackenbush J, Irizarry RA, Bravo HC (2018). <a href="https://academic.oup.com/biostatistics/article-abstract/19/2/185/3949169">Smooth quantile normalization</a>. <em>Biostatistics</em>.</p>
<p>Kumar MS, Slud EV, Okrah K, <strong>Hicks SC</strong>, Hannenhalli S, Corrada Bravo H (2018). <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-5160-5">Analysis and correction of compositional bias in sparse sequencing count data</a>. <em>BMC Genomics</em>.</p>
<p><strong>Hicks SC</strong>, Irizarry RA (2018). <a href="https://www.tandfonline.com/doi/abs/10.1080/00031305.2017.1356747?journalCode=utas20">A Guide to Teaching Data Science</a>. <em>The American Statistician</em>.</p>
<p><strong>Hicks SC</strong> (2017). <a href="https://www.tandfonline.com/doi/abs/10.1080/10618600.2017.1385472">Greater Data Science Ahead</a>. <em>Journal of Computational Graphical Statistics</em>.</p>
<p><strong>Hicks SC</strong>, Irizarry RA (2015). <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0679-0">quantro: a data-driven approach to guide the choice of an appropriate normalization method</a>. <em>Genome Biology</em>.</p>
<p>Cheung HC, San Lucas FA, <strong>Hicks S</strong>, Chang K, Bertuch AA, Ribes-Zamora A. (2012). <a href="https://pubmed.ncbi.nlm.nih.gov/23176708/">An S/T-Q cluster domain census unveils new putative targets under Tel1/Mec1 control</a>. <em>BMC Genomics</em>.</p>
<p><strong>Hicks S</strong>, Wheeler DE, Plon SE, Kimmel M. (2011). <a href="https://pubmed.ncbi.nlm.nih.gov/21480434/">Prediction of Missense Mutation Functionality Depends on both the Algorithm and Sequence Alignment Employed</a>. <em>Human Mutation</em>.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/www\.stephaniehicks\.com");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>