---
title: "Publications"
format:
    html:
        toc: true
        toc-depth: 3
---

The following are a selected set of publications. To see a list with all of papers from Stephanie Hicks, you can visit [Google Scholar](https://scholar.google.com/citations?user=4T4qnL4AAAAJ&hl=en) or [ORCiD](https://orcid.org/0000-0002-7858-0231).

::: callout-tip
-   The $^*$ indicates a mentored student or post-doctoral fellow.
-   The $\dagger$ and $\ddagger$ indicate co-first and co-last authors, respectively.
:::

## Preprints

Shah K$^*$, Totty MS$^*$, Bach SV, Valentine MR, Chandra A, Divecha HR, Miller RA, Kwon SH, Ramnauth AD, Tippani M, Tyagi S, Kleinman JE, Collado-Torres L, Han S, Hyde TM, Page SC, Maynard KR, **Hicks SC**$^{\dagger}$, Martinowich K$^{\dagger}$. (2025). [Spatio-molecular gene expression reflects dorsal anterior cingulate cortex structure and function in the human brain](https://doi.org/10.1101/2025.07.14.664821). *bioRxiv*.

Mulvey B, Wang Y, Divecha HR, Bach SV, Montgomery KD, Cinquemani S, Chandra A, Du Y, Miller RA, Kleinman JE, Page SC, Hyde TM, Martinowich K, **Hicks SC**, Hansen KD$^{\ddagger}$, Maynard KR$^{\ddagger}$. (2024). [Spatially-resolved molecular sex differences at single cell resolution in the adult human hypothalamus](https://doi.org/10.1101/2024.12.07.627362) *bioRxiv*.

Wang, A, **Hicks SC**, Geman D, Younes L. (2024). [GeneCover: A Combinatorial Approach for Label-free Marker Gene Selection](https://doi.org/10.1101/2024.10.30.621151). *bioRxiv*. (Accepted for and presented at [RECOMB 2025](https://recomb.org/recomb2025/)).

Peng RD, **Hicks SC**. (2024). [Modeling Data Analytic Iteration With Probabilistic Outcome Sets](https://arxiv.org/abs/2309.08494). *arXiv*.

Hippen AA, Davidson NR, Barnard ME, Weber LM$^{*}$, Gertz J, Doherty JA, **Hicks SC**, Greene CS. (2023). [Deconvolution reveals compositional differences in high-grade serous ovarian cancer subtypes](https://doi.org/10.1101/2023.06.14.544991). *bioRxiv*.

D'Agostino McGowan L, Peng RD, **Hicks SC**. (2023). [Quantifying the Alignment of a Data Analysis between Analyst and Audience](https://arxiv.org/abs/2312.07616). *arXiv*.

**Hicks SC**, Peng RD. (2019). [Elements and Principles for Characterizing Variation between Data Analyses](https://arxiv.org/abs/1903.07639). *arXiv*.

## 2025

Thompson JR$^{*,\dagger}$, Nelson ED$^{*,\dagger}$, Tippani M$^{\dagger}$, Ramnauth AD, Divecha HR, Miller RA, Eagles NJ, Pattie EA, Kwon SH, Bach SV, Kaipa UM, Yao J$^{*}$, Hou C$^{*}$, Kleinman JE, Collado-Torres L, Han S, Maynard KR, Hyde TM, Martinowich K, Page SC$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2025). [An integrated single-nucleus and spatial transcriptomics atlas reveals the molecular landscape of the human hippocampus](https://doi.org/10.1101/2024.04.26.590643). Accepted in *Nature Neuroscience*.

Totty MS$^{*,\dagger}$, Cervera-Juanes R$^{\dagger}$, Bach SV, Ameur LB, Valentine MR, Simons E, Romac MD, Trinh H, Henderson K, Rosario ID, Tippani M, Miller RA, Kleinman JE, Page SC, Saunders A, Hyde TM, Martinowich K$^{\ddagger}$, **Hicks SC**$^{\ddagger}$, Costa VD$^{\ddagger}$. (2024). [Transcriptomic diversity of amygdalar subdivisions across humans and nonhuman primates](https://doi.org/10.1101/2024.10.18.618721). Accepted in *Science Advances*.

Zhou H$^{*}$, Panwar P, Guo B$^{*}$, Hallinan C$^{*}$, Ghazanfar S$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2025). [Spatial mutual nearest neighbors for spatial transcriptomics data](https://doi.org/10.1093/bioinformatics/btaf403). *Bioinformatics*.

Luecken MD, Gigante S, Burkhardt DB, Cannoodt R, Strobl DC, Markov NS, Zappia L, Palla G, Lewis W, Dimitrov D, Vinyard ME, Magruder DS, Andersson A, Dann E, Qin Q, Otto DJ, Klein M, Botvinnik OB, Deconinck L, Waldrant K, **The Open Problems Jamboree Members**, Bloom JM, Pisco AO, Saez-Rodriguez J, Wulsin D, Pinello L, Saeys Y, Theis FJ, Krishnaswamy S. (2025). [Defining and benchmarking open problems in single-cell analysis](https://doi.org/10.1038/s41587-025-02694-w). *Nature Biotechnology*.

Totty MS$^{*}$, **Hicks SC**, Guo B$^{*}$. (2025). [SpotSweeper: spatially-aware quality control for spatial transcriptomics](https://doi.org/10.1038/s41592-025-02713-3). *Nature Methods*.

Yao J$^{*}$, Yu J, Caffo B, Page SC, Martinowich K, **Hicks SC**. (2025). [Spatial domain detection using contrastive self-supervised learning for spatial multi-omics technologies](https://doi.org/10.1101/gr.279380.124). *Genome Reserach*.

Popescu A, Ottaway C, Ford K, Patterson TW, Ingiosi AM, Medina E, **Hicks SC**, Singletary K, Peixoto L. (2025). [Transcriptional Dynamics of Sleep Deprivation and Subsequent Recovery Sleep in the Male Mouse Cortex](https://journals.physiology.org/doi/abs/10.1152/physiolgenomics.00128.2024). *Physiological Genomics*.

Maden SK$^{*}$, Huuki-Myers LA, Kwon SH, Collado-Torres L, Maynard KR, **Hicks SC**. (2025). [lute: estimating the cell composition of heterogeneous tissue with varying cell sizes using gene expression](https://doi.org/10.1186/s12864-025-11508-x). *BMC Genomics*.

Ivich A, Davidson NR, Grieshober L, Li W$^{*}$, **Hicks SC**, Doherty JA, Greene CS. (2025). [Missing cell types in single-cell references impact deconvolution of bulk data but are detectable](https://doi.org/10.1186/s13059-025-03506-9). *Genome Biology*.

Shah K$^{*}$, Guo B$^{*}$, **Hicks SC**. (2025). [Addressing the mean-variance relationship in spatially resolved transcriptomics data with *spoon*](https://doi.org/10.1093/biostatistics/kxaf012). *Biostatistics*.

Huuki-Myers LA, Montgomery KD, Kwon SH, Cinquemani S, Maden SK$^{*}$, Eagles NJ, Kleinman JE, Hyde TM, **Hicks SC**, Maynard KR$^{\ddagger}$, Collado-Torres L$^{\ddagger}$. (2025). [Benchmark of cellular deconvolution methods using a multi-assay reference dataset from postmortem human prefrontal cortex](https://doi.org/10.1186/s13059-025-03552-3). *Genome Biology*.

Greene CS, Gignoux CR, Subirana-Gran√©s M, Pividori M, **Hicks SC**, Ackert-Bicknell CL. (2025). [Can AI reveal the next generation of high-impact bone genomics targets?](https://www.sciencedirect.com/science/article/pii/S2352187225000166). *Bone Reports*.

Hou C$^{*}$, Ghazanfar S, Marini F, Morgan M, **Hicks SC**. (2025). [HuBMAPR: an R Client for the HuBMAP Data Portal](https://doi.org/10.1093/bioadv/vbaf048). *Bioinformatics Advances*.

Ramnauth AD$^{\dagger}$, Tippani M, Divecha HR, Papariello AR, Miller RA, Nelson ED$^{*}$, Thompson JR$^{*}$, Pattie EA, Kleinman JE, Maynard KR, Collado-Torres L, Hyde TM, Martinowich K$^{\ddagger}$, **Hicks SC**$^{\ddagger}$, Page SC$^{\ddagger}$. (2025). [Spatially-resolved transcriptomics of human dentate gyrus across postnatal lifespan reveals heterogeneity in markers for proliferation, extracellular matrix, and neuroinflammation](https://doi.org/10.1016/j.celrep.2025.115300). *Cell Reports*.

Guo B$^{*,\dagger}$, Ling W$^{\dagger}$, Kwon SH, Panwar P, Ghazanfar S$^{\ddagger}$, Martinowich K$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2025). [Integrating spatially-resolved transcriptomics data across tissues and individuals: challenges and opportunities](https://doi.org/10.1002/smtd.202401194). *Small Methods*.

Phillips III RA, Bach SV, Du Y, Miller RA, Kleinman JE, Hyde TM, **Hicks SC**$^{\ddagger}$, Page SC$^{\ddagger}$, Martinowich K$^{\ddagger}$. (2025). [Transcriptomic characterization of human lateral septum neurons reveals conserved and divergent marker genes across species](https://doi.org/10.1016/j.isci.2025.111820). *iScience*.

## 2024

Wright C, Meng Q, Breshock MR, Atta L, Taub MA, Jager LR, Muschelli J, **Hicks SC** (2024). [Open Case Studies: Statistics and Data Science Education through Real-World Applications](https://doi.org/10.1080/26939169.2024.2394541). *Journal of Statistics and Data Science Education*.

Ford K, Zuin E, Righelli D, Medina E, Schoch H, Singletary K, Muheim C, Frank MG, **Hicks SC**, Risso D, Peixoto L. (2024). [A global transcriptional atlas of the effect of sleep deprivation in the mouse frontal cortex.](https://www.cell.com/iscience/fulltext/S2589-0042(24)01977-1). *iScience*.

Hutchison WJ, Keyes TJ, The Tidyomics Consortium, Crowell LH, Soneson C, Yuan V, Nahid AA, Mu W, Park J, Davis ES, Tang M, Axisa PP, Kitt W, Poon CL, Kosmac M, Serizay J, Sato N, Gottardo R, Morgan M, Lee S, Lawrence M, **Hicks SC**, Nolan GP, Davis KL, Papenfuss AT$^{\ddagger}$, Love M$^{\ddagger}$, Mangiola S$^{\ddagger}$. (2024). [The tidyomics ecosystem: Enhancing omic data analyses](https://doi.org/10.1038/s41592-024-02299-2). *Nature Methods*.

Huuki-Myers LA, Spangler A, Eagles NJ, Montgomery KD, Kwon SH, Guo B$^{*}$, Grant-Peters M, Divecha HR, Tippani M, Sriworarat C$^{*}$, Nguyen AB, Ravichandran P$^{*}$, Tran MT, Seyedian A, PsychENCODE Consortium, Hyde TM, Kleinman JE, Battle A, Page SC, Ryten M, **Hicks SC**, Martinowich K, Collado-Torres L$^{\ddagger}$, Maynard KR$^{\ddagger}$. (2024). [A data-driven single-cell and spatial transcriptomic map of the human prefrontal cortex](https://www.science.org/doi/10.1126/science.adh1938). *Science*.

Ruzicka WB, Mohammadi S, Fullard JF, Davila-Velderrain J, Subburaju S, Tso DR, Hourihan M, Jiang S, Lee H-C, Bendl J, **PsychENCODE Consortium**, Voloudakis G, Haroutunian V, Hoffman GE, Roussos P, Kellis M. (2024). [Single-cell multi-cohort dissection of the schizophrenia transcriptome](https://www.science.org/doi/10.1126/science.adg5136). *Science*.

Weber LM$^{*,\dagger}$, Divecha HR$^{\dagger}$, Tran MN, Kwon SH, Spangler A, Montgomery KD, Tippani M, Bharadwaj R, Kleinman JE, Page SC, Hyde TM, Collado-Torres L, Maynard KR, Martinowich K$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2024). [The gene expression landscape of the human locus coeruleus revealed by single-nucleus and spatially-resolved transcriptomics](https://elifesciences.org/articles/84628). *eLife*.

Erdogdu B, Varabyou A, **Hicks SC**, Salzberg SL, Pertea M. (2024). [Detecting differential transcript usage in complex diseases with SPIT](https://pubmed.ncbi.nlm.nih.gov/38508189/). *Cell Reports Methods*.

Kuo A$^{*}$, Hansen KD, **Hicks SC**. (2024). [Quantification and statistical modeling of Chromium-based single-nucleus RNA-sequencing data](https://doi.org/10.1093/biostatistics/kxad010). *Biostatistcs*.

## 2023

Weber LM$^{*}$, Saha A, Datta A, Hansen KD, **Hicks SC**. (2023). [nnSVG: scalable identification of spatially variable genes using nearest-neighbor Gaussian processes](https://doi.org/10.1038/s41467-023-39748-z). *Nature Communications*.

Guo B$^{*}$, Huuki-Myers LA, Grant-Peters M, Collado-Torres L, **Hicks SC**. (2023). [escheR: Unified multi-dimensional visualizations with Gestalt principles](https://doi.org/10.1093/bioadv/vbad179). *Bioinformatics Advances*.

Sriworarat C$^{*}$, Nguyen AB, Eagles NJ, Collado-Torres L, Martinowich K, Maynard KR$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2023). [Performant web-based interactive visualization tool for spatially-resolved transcriptomics experiments](https://doi.org/10.1017/S2633903X2300017X). *Biological Imaging*.

Tippani M, Divecha HR, Catallini JL, Weber LM$^{*}$, Spangler A, Jaffe AE, **Hicks SC**, Martinowich K, Collado-Torres L, Page SC$^{\ddagger}$, Maynard KR$^{\ddagger}$. (2023). [VistoSeg: processing utilities for high-resolution Visium/Visium-IF images for spatial transcriptomics data](https://doi.org/10.1017/S2633903X23000235). *Biological Imaging*.

Kwon SH, Parthiban S$^{*}$, Tippani M, Divecha HR, Eagles NJ, Lobana JS, Williams SR, Mark M, Bharadwaj RA, Kleinman JE, Hyde TM, Page SC, **Hicks SC**$^{\ddagger}$, Martinowich K$^{\ddagger}$, Maynard KR$^{\ddagger}$, Collado-Torres L$^{\ddagger}$. (2023). [Influence of Alzheimer's disease related neuropathology on local microenvironment gene expression in the human inferior temporal cortex](https://doi.org/10.1089/genbio.2023.0019). *GEN Biotechnology*.

Maden SK$^{*}$, Kwon SH, Huuki-Myers LA, Collado-Torres L, **Hicks SC**$^{\ddagger}$, Maynard KR$^{\ddagger}$. (2023). [Challenges and opportunities to computationally deconvolve heterogeneous tissue with varying cell sizes using single cell RNA-sequencing datasets](https://doi.org/10.1186/s13059-023-03123-4). *Genome Biology*.

Hou W$^{*}$, Ji Z, Chen Z, Wherry EJ, **Hicks SC**$^{\ddagger}$, Ji H$^{\ddagger}$. (2023). [A statistical framework for differential pseudotime analysis with multiple single-cell RNA-seq samples](https://doi.org/10.1038/s41467-023-42841-y). *Nature Communications*.

Hippen AA, Omran DK, Weber LM$^{*}$, Jung E, Drapkin R, Doherty JA, **Hicks SC**, Greene CS. (2023). [Performance of computational algorithms to deconvolve heterogeneous bulk tumor tissue depends on experimental factors](https://doi.org/10.1186/s13059-023-03077-7). *Genome Biology*.

Nelson ED$^{*}$, Maynard KR, Nicholas KR, Tran MN, Divecha HR, Collado-Torres L, **Hicks SC**$^{\ddagger}$, Martinowich K$^{\ddagger}$. (2023). [Activity-regulated gene expression across cell types of the mouse hippocampus](https://doi.org/10.1002/hipo.23548). *Hippocampus*.

Dyjack N$^{*}$, Baker DN, Braverman V, Langmead B, **Hicks SC**. (2023). [A scalable and unbiased discordance metric with H+](https://pubmed.ncbi.nlm.nih.gov/36063544/). *Biostatistics*.

Huuki-Myers LA, Montgomery KD, Kwon SH, Page SC, **Hicks SC**, Maynard KR, Collado-Torres L. (2023). [Data-driven Identification of Total RNA Expression Genes (TREGs) for Estimation of RNA Abundance in Heterogeneous Cell Types](https://doi.org/10.1186/s13059-023-03066-w). *Genome Biology*.

## 2022

Righelli D$^{\dagger}$, Weber LM$^{*,\dagger}$, Crowell HL$^{\dagger}$, Pardo B, Collado-Torres L, Ghazanfar S, Lun ATL, **Hicks SC**$^{\ddagger}$, Risso D$^{\ddagger}$ (2022). [SpatialExperiment: infrastructure for spatially resolved transcriptomics data in R using Bioconductor](https://doi.org/10.1093/bioinformatics/btac299). *Bioinformatics*.

Wang Y, **Hicks SC**, Hansen KD (2022). [Addressing the mean-correlation relationship in co-expression analysis](https://doi.org/10.1371/journal.pcbi.1009954). *PLOS Computational Biology*.

D'Agostino McGowan L, Peng RD, **Hicks SC**. (2022). [Design Principles for Data Analysis](https://doi.org/10.1080/10618600.2022.2104290). *Journal of Computational and Graphical Statistics*.

Pardo B, Spangler A, Weber LM$^{*}$, **Hicks SC**, Jaffe AE, Martinowich K, Maynard KR, Collado-Torres L. (2022). [spatialLIBD: an R/Bioconductor package to visualize spatially-resolved transcriptomics data](https://doi.org/10.1186/s12864-022-08601-w). *BMC Genomics*.

Mallick H, Chatterjee Su., Chowdhury S, Chatterjee Sa., Rahnavard A$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2022). [Differential expression of single-cell RNA-seq data using Tweedie models](http://doi.org/10.1002/sim.9430). *Statistics in Medicine*.

## 2021

Heil BJ, Hoffman MM, Markowetz F, Lee S-I, Greene CS$^{\ddagger}$, **Hicks SC**$^{\ddagger}$. (2021). [Reproducibility standards for machine learning in the life sciences](https://doi.org/10.1038/s41592-021-01256-7). *Nature Methods*.

Maynard KR$^{\dagger}$, Collado-Torres L$^{\dagger}$, Weber LM$^{*}$, Uytingco C, Barry BK, Williams SR, II JLC, Tran MN, Besich Z, Tippani M, Chew J, Yin Y, Kleinman JE, Hyde TM, Rao N, **Hicks SC**, Martinowich K$^{\ddagger}$, Jaffe AE$^{\ddagger}$ (2021). [Transcriptome-scale spatial gene expression in the human dorsolateral prefrontal cortex](https://doi.org/10.1038/s41593-020-00787-0). *Nature Neuroscience*.

Hippen AA$^{*}$, Falco MM, Weber LM$^{*}$, Erkan EP, Zhang K, Dohert JA, V√§h√§rautio A$^{\ddagger}$, Greene CS, **Hicks SC**$^{\ddagger}$. (2021). [miQC: An adaptive probabilistic framework for quality control of single-cell RNA-sequencing data](https://doi.org/10.1371/journal.pcbi.1009290). *PLOS Computational Biology*.

Baker DN, Dyjack N$^{*}$, Braverman V, **Hicks SC**, Langmead B. (2021). [minicore: Fast scRNA-seq clustering with various distance measures](https://doi.org/10.1145/3459930.3469523). *ACM Conference on Bioinformatics, Computational Biology and Biomedicine*.

Weber LM$^{*}$, Hippen AA, Hickey PF, Berrett KC, Gertz J, Doherty JA, Greene CS, **Hicks SC**. (2021). [Genetic demultiplexing of pooled single-cell RNA-sequencing samples in cancer facilitates effective experimental design](https://doi.org/10.1093/gigascience/giab062). *GigaScience*.

Tran MN$^{\dagger}$, Maynard KR$^{\dagger}$, Spangler A, Collado-Torres L, Sadashivaiah V, Tippani M, Barry BK, Hancock DB, **Hicks SC**, Kleinman JE, Hyde TM, Martinowich K$^{\ddagger}$, Jaffe A$^{\ddagger}$. (2021). [Single-nucleus transcriptome analysis reveals cell type-specific molecular signatures across reward circuitry in the human brain](https://doi.org/10.1016/j.neuron.2021.09.001). *Neuron*.

Peng RD, Chen A, Bridgeford E, Leek JT, and **Hicks SC**. (2021). [Diagnosing Data Analytic Problems in the Classroom](https://doi.org/10.1080/26939169.2021.1971586). *Journal of Statistics and Data Science Education*.

Peng RD, **Hicks SC**. (2021). [Reproducible Research: A Retrospective](https://doi.org/10.1146/annurev-publhealth-012420-105110). *Annual Review of Public Health*.

**Hicks SC**, Liu R, Ni Y, Purdom E, Risso D. (2021). [mbkmeans: fast clustering for single cell data using mini-batch *k*-means](https://doi.org/10.1371/journal.pcbi.1008625). *PLOS Computational Biology*.

## Earlier years

Amezquita RA, Lun ATL, Carey VJ, Carpp LN, Geistlinger L, Marini F, Rue-Albrecht K, Risso D, Soneson C, Waldron L, Pages H, Smith M, Huber W, Morgan M, Gottardo R, **Hicks SC**. (2020). [Orchestrating Single-Cell Analysis with Bioconductor](https://doi.org/10.1038/s41592-019-0654-x). *Nature Methods*.

Lahnemann D, Koester J, Szczurek E, McCarthy D, **Hicks S** *et al*. (2020). [Eleven grand challenges in single-cell data science](https://pubmed.ncbi.nlm.nih.gov/32033589/). *Genome Biology*.

Hou W$^{*}$, Ji J, Ji H, **Hicks SC** (2020). [A systematic evaluation of single-cell RNA-sequencing imputation methods](https://doi.org/10.1186/s13059-020-02132-x). *Genome Biology*.

Huang R, Soneson C, Ernst FGM, Rue-Albrecht KC, Guangchuang Y, **Hicks SC**, Robinson MD. (2020). [TreeSummarizedExperiment: a S4 class for data with hierarchical structure](https://doi.org/10.12688/f1000research.26669.1). *F1000Research*.

Townes FW, **Hicks SC**, Aryee MJ, Irizarry RA (2019). [Feature Selection and Dimension Reduction for Single Cell RNA-Seq based on a Multinomial Model](https://doi.org/10.1186/s13059-019-1861-6). *Genome Biology*.

**Hicks SC**, Irizarry RA. (2019). [methylCC: technology-independent estimation of cell type composition using differentially methylated regions](https://doi.org/10.1186/s13059-019-1827-8). *Genome Biology*.

Korthauer K$^{\dagger}$, Kimes PK$^{\dagger}$, Duvallet C, Reyes A, Subramanian A, Teng M, Shukla C, Alm EJ, **Hicks SC** (2019). [A practical guide to methods controlling false discoveries in computational biology](https://doi.org/10.1186/s13059-019-1716-1). *Genome Biology*.

**Hicks SC**, Townes FW, Teng M, Irizarry RA (2018). [Missing data and technical variability in single-cell RNA-sequencing experiments](https://academic.oup.com/biostatistics/article/19/4/562/4599254). *Biostatistics*.

**Hicks SC**, Okrah K, Paulson JN, Quackenbush J, Irizarry RA, Bravo HC (2018). [Smooth quantile normalization](https://academic.oup.com/biostatistics/article-abstract/19/2/185/3949169). *Biostatistics*.

Kumar MS, Slud EV, Okrah K, **Hicks SC**, Hannenhalli S, Corrada Bravo H (2018). [Analysis and correction of compositional bias in sparse sequencing count data](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-5160-5). *BMC Genomics*.

**Hicks SC**, Irizarry RA (2018). [A Guide to Teaching Data Science](https://www.tandfonline.com/doi/abs/10.1080/00031305.2017.1356747?journalCode=utas20). *The American Statistician*.

**Hicks SC** (2017). [Greater Data Science Ahead](https://www.tandfonline.com/doi/abs/10.1080/10618600.2017.1385472). *Journal of Computational Graphical Statistics*.

**Hicks SC**, Irizarry RA (2015). [quantro: a data-driven approach to guide the choice of an appropriate normalization method](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0679-0). *Genome Biology*.

Cheung HC, San Lucas FA, **Hicks S**, Chang K, Bertuch AA, Ribes-Zamora A. (2012). [An S/T-Q cluster domain census unveils new putative targets under Tel1/Mec1 control](https://pubmed.ncbi.nlm.nih.gov/23176708/). *BMC Genomics*.

**Hicks S**, Wheeler DE, Plon SE, Kimmel M. (2011). [Prediction of Missense Mutation Functionality Depends on both the Algorithm and Sequence Alignment Employed](https://pubmed.ncbi.nlm.nih.gov/21480434/). *Human Mutation*.