<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Software and data packages – Stephanie C. Hicks, PhD</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-587c61ba64f3a5504c4d52d930310e48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-960d9d362f81d399278ec3e29b622b37.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Stephanie C. Hicks, PhD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./projects.html"> 
<span class="menu-text">Research themes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./people.html"> 
<span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./publications.html"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./software.html" aria-current="page"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./teaching.html"> 
<span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./talks.html"> 
<span class="menu-text">Talks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./join.html"> 
<span class="menu-text">Join us</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/stephaniehicks"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/stephaniehicks"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#bioconductor" id="toc-bioconductor" class="nav-link active" data-scroll-target="#bioconductor">Bioconductor</a></li>
  <li><a href="#cran" id="toc-cran" class="nav-link" data-scroll-target="#cran">CRAN</a></li>
  <li><a href="#python" id="toc-python" class="nav-link" data-scroll-target="#python">Python</a></li>
  <li><a href="#github" id="toc-github" class="nav-link" data-scroll-target="#github">GitHub</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Software and data packages</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="bioconductor" class="level2">
<h2 class="anchored" data-anchor-id="bioconductor">Bioconductor</h2>
<section id="software-packages" class="level4">
<h4 class="anchored" data-anchor-id="software-packages">Software packages</h4>
<ul>
<li><a href="https://bioconductor.org/packages/HuBMAPR">R/HuBMAPR</a> | <img src="https://bioconductor.org/shields/years-in-bioc/HuBMAPR.svg"> | R/Bioconductor package that serves as an R Client for the HuBMAP Data Portal. (<a href="https://doi.org/10.1093/bioadv/vbaf048">Hou et al., 2025. <em>Bioinformatics Advances</em></a>)</li>
<li><a href="https://bioconductor.org/packages/spoon">R/spoon</a> | <img src="https://bioconductor.org/shields/years-in-bioc/spoon.svg"> | R/Bioconductor package that addresses the mean-variance relationship in spatially resolved transcriptomics data. (<a href="https://doi.org/10.1093/biostatistics/kxaf012">Shah et al., 2025. <em>Biostatistics</em></a>).</li>
<li><a href="https://bioconductor.org/packages/SpotSweeper">R/SpotSweeper</a> | <img src="https://bioconductor.org/shields/years-in-bioc/SpotSweeper.svg"> | R/Bioconductor package that provides spatially-aware quality control metrics for spatial transcriptomics data. (<a href="https://doi.org/10.1038/s41592-025-02713-3">Totty et al., 2025. <em>Nature Methods</em></a>).</li>
<li><a href="https://bioconductor.org/packages/escheR">R/escheR</a> | <img src="https://bioconductor.org/shields/years-in-bioc/escheR.svg"> | Built off of ggplot2 and the Gestalt principles to visualize multi-dimensional data in the 2D space (e.g.&nbsp;embedding or spatial visualizations). (<a href="https://doi.org/10.1093/bioadv/vbad179">Guo et al., 2023. <em>Bioinformatics Advances</em></a>).</li>
<li><a href="https://www.bioconductor.org/packages/TREG">R/TREG</a> | <img src="https://bioconductor.org/shields/years-in-bioc/TREG.svg"> | R/Bioconductor package to find Total RNA Expression Genes (TREGs) in single nucleus RNA-seq data (<a href="https://doi.org/10.1186/s13059-023-03066-w">Huuki-Myers et al., 2023. <em>Genome Biology</em></a>).</li>
<li><a href="https://www.bioconductor.org/packages/nnSVG">R/nnSVG</a> | <img src="https://bioconductor.org/shields/years-in-bioc/nnSVG.svg"> | R/Bioconductor package that provides a scalable method to identify spatially variable genes (SVGs) in spatially-resolved transcriptomics data based on nearest-neighbor Gaussian processes. (<a href="https://doi.org/10.1038/s41467-023-39748-z">Weber et al., 2023. <em>Nature Communications</em></a>).</li>
<li><a href="https://www.bioconductor.org/packages/miQC">R/miQC</a> | <img src="https://bioconductor.org/shields/years-in-bioc/miQC.svg"> | R/Bioconductor package with a data-driven quality control metric to predict the low-quality scRNA-seq cells. (<a href="https://doi.org/10.1371/journal.pcbi.1009290">Hippen et al., 2021. <em>PLOS Computational Biology</em></a>).</li>
<li><a href="https://www.bioconductor.org/packages/SpatialExperiment">R/SpatialExperiment</a> | <img src="https://bioconductor.org/shields/years-in-bioc/SpatialExperiment.svg"> | R/Bioconductor package to define a S4 class for data with spatial coordinates by extending SingleCellExperiment to include spatial experiments such as seqFISH and 10x Visium spatial transcriptomics technologies. (<a href="https://doi.org/10.1093/bioinformatics/btac299">Righelli, Weber, Crowell et al., 2021. <em>Bioinformatics</em></a>).</li>
<li><a href="https://www.bioconductor.org/packages/bluster">R/bluster</a> | <img src="https://bioconductor.org/shields/years-in-bioc/bluster.svg"> | R/Bioconductor package to wrap common clustering algorithms in an easily extended S4 framework. Backends are implemented for hierarchical, k-means and graph-based clustering. Several utilities are also provided to compare and evaluate clustering results.</li>
<li><a href="http://bioconductor.org/packages/spqn">R/spQN</a> | <img src="https://bioconductor.org/shields/years-in-bioc/spqn.svg"> | R/Bioconductor package to implement spatial quantile normalization (SpQN). This method was developed to remove a mean-correlation relationship in correlation matrices built from gene expression data. (<a href="https://doi.org/10.1371/journal.pcbi.1009954">Wang et al., 2020. <em>PLoS Computational Biology</em></a>).</li>
<li><a href="https://bioconductor.org/packages/scry">R/scry</a> | <img src="https://bioconductor.org/shields/years-in-bioc/scry.svg"> | R/Bioconductor package to implement count-based feature selection and dimension reduction algorithms to facilitate unsupervised analysis of high-dimensional data, such as single-cell RNA-seq. This package builds around <a href="https://cran.r-project.org/web/packages/glmpca/index.html">glmpca R CRAN package</a>. (<a href="https://doi.org/10.1186/s13059-019-1861-6">Townes et al., 2019. <em>Genome Biology</em></a>).</li>
<li><a href="http://bioconductor.org/packages/methylCC">R/methylCC</a> | <img src="https://bioconductor.org/shields/years-in-bioc/methylCC.svg"> | R/Bioconductor package to estimate the cell composition of whole blood in DNA methylation samples. (<a href="https://doi.org/10.1186/s13059-019-1827-8">Hicks and Irizarry, 2019. <em>Genome Biology</em></a>).</li>
<li><a href="http://bioconductor.org/packages/mbkmeans">R/mbkmeans</a> | <img src="https://bioconductor.org/shields/years-in-bioc/mbkmeans.svg"> | R/Bioconductor package implementing the mini-batch optimization for <em>k</em>-means (<strong>mbkmeans</strong>)clustering proposed in <a href="https://www.eecs.tufts.edu/~dsculley/papers/fastkmeans.pdf">Sculley (2010)</a> for large datasets, including scRNA-seq data. The mini batch <em>k</em>-means algorithm can be run with data stored in-memory or on-disk (e.g.&nbsp;HDF5 file format). (<a href="https://doi.org/10.1371/journal.pcbi.1008625">Hicks et al., 2021. <em>PLoS Computational Biology</em></a>).</li>
<li><a href="https://www.bioconductor.org/packages/TreeSummarizedExperiment">R/TreeSummarizedExperiment</a> | <img src="https://bioconductor.org/shields/years-in-bioc/TreeSummarizedExperiment.svg"> | R/Bioconductor package to define a S4 class for data with hierarchical structure by extending SingleCellExperiment to include hierarchical information on the rows or columns of the rectangular data. (<a href="https://doi.org/10.12688/f1000research.26669.1">Huang et al., 2020. <em>F1000</em></a>).</li>
<li><a href="http://bioconductor.org/packages/qsmooth">R/qsmooth</a> | <img src="https://bioconductor.org/shields/years-in-bioc/qsmooth.svg"> | R/Bioconductor package available that implements a generalization of quantile normalization, referred to as smooth quantile normalization (<strong>qsmooth</strong>), which is based on the assumption that the statistical distribution of each sample should be the same (or have the same distributional shape) within biological groups or conditions. (<a href="https://doi.org/10.1093/biostatistics/kxx028">Hicks et al., 2018. <em>Biostatistics</em></a>).</li>
<li><a href="http://www.bioconductor.org/packages/release/bioc/html/quantro.html">R/quantro</a> | <img src="https://bioconductor.org/shields/years-in-bioc/quantro.svg"> | R/Bioconductor package to test for global differences between groups of distributions to decide when to use quantile normalization. (<a href="https://doi.org/10.1186/s13059-015-0679-0">Hicks and Irizarry, 2015. <em>Genome Biology</em></a>).</li>
</ul>
</section>
<section id="data-packages" class="level4">
<h4 class="anchored" data-anchor-id="data-packages">Data packages</h4>
<ul>
<li><a href="https://www.bioconductor.org/packages/humanHippocampus2024">humanHippocampus2024</a> | Access spatially-resolved transcriptomics and single-nucleus RNA sequencing data from adjacent tissue sections of the anterior human hippocampus across ten adult neurotypical donors from <a href="https://doi.org/10.1038/s41593-025-02022-0">Thompson, Nelson, Tippani et al.&nbsp;(2025). <em>Nature Neuroscience</em></a>.</li>
<li><a href="https://www.bioconductor.org/packages/STexampleData">STexampleData</a> | Collection of spatially resolved transcriptomics datasets in SpatialExperiment Bioconductor format, for use in examples, demonstrations, tutorials, and other purposes.</li>
<li><a href="http://www.bioconductor.org/packages/spatialLIBD">spatialLIBD</a> | Inspect interactively spatial transcriptomics 10x Genomics Visium data in postmorteum human DLPFC tissue. (<a href="https://doi.org/10.1038/s41593-020-00787-0">Maynard, Collado-Torres et al.&nbsp;(2021). <em>Nature Neuroscience</em></a>).</li>
<li><a href="http://bioconductor.org/packages/benchmarkfdrData2019">benchmarkfdrData2019</a> | Benchmarking results and data sets used in <a href="https://doi.org/10.1186/s13059-019-1716-1">Korthauer and Kimes et al.&nbsp;(2019)</a> to compare methods for controlling the false discovery rate. A shiny app to interactively explore the data is available on the <a href="https://github.com/kdkorthauer/benchmarkfdr-shiny">benchmark-shiny repository</a> on GitHub.</li>
<li><a href="http://bioconductor.org/packages/bodymapRat">bodymapRat</a> | R data package that contains an SummarizedExperiment from the <a href="https://www.ncbi.nlm.nih.gov/pubmed/24510058">Yu et al.&nbsp;(2013)</a> paper that performed the rat BodyMap across 11 organs and 4 developmental stage.</li>
<li><a href="http://bioconductor.org/packages/TENxPBMCData">TENxPBMCData</a> | R data package that contains a set of SingleCellExperiment objects with single-cell RNA-sequencing data from peripheral blood mononuclear cells generated by 10X Genomics.</li>
</ul>
</section>
<section id="books" class="level4">
<h4 class="anchored" data-anchor-id="books">Books</h4>
<ul>
<li><a href="https://bioconductor.org/books/release/OSTA">Orchestrating Spatial Transcriptomics Analysis with Bioconductor</a> (OSTA). (<a href="https://doi.org/10.1101/2025.11.20.688607">Crowell and Dong et al., 2025. <em>bioRxiv</em></a>)</li>
<li><a href="http://bioconductor.org/books/release/OSCA">Orchestrating Single-Cell Analysis in Bioconductor</a> (<a href="https://doi.org/10.1038/s41592-019-0654-x">Amezquita et al., 2020. <em>Nature Methods</em></a>)</li>
</ul>
</section>
</section>
<section id="cran" class="level2">
<h2 class="anchored" data-anchor-id="cran">CRAN</h2>
<section id="software-packages-1" class="level4">
<h4 class="anchored" data-anchor-id="software-packages-1">Software packages</h4>
<ul>
<li><a href="https://cran.r-project.org/web/packages/fasthplus">R/fasthplus</a> | Provides fast approximations for metrics of discordance or dissimilarity including (1) to evaluate the discordance between two arbitrary sets or (2) to evaluate label fitness (clustering) for a generalized dissimilarity matrix. (<a href="https://doi.org/10.1093/biostatistics/kxac035">Dyjack et al., 2022. <em>Biostatistics</em></a>).</li>
<li><a href="https://cran.r-project.org/web/packages/OCSdata">R/OCSdata</a> | Provides functions to access and download data from the <a href="https://www.opencasestudies.org">Open Case Studies</a> repositories on <a href="https://github.com/opencasestudies">GitHub</a>. Different functions enable users to grab the data they need at different sections in the case study, as well as download the whole case study repository.</li>
<li><a href="https://cran.r-project.org/web/packages/glmpca/index.html">R/glmpca</a> | Implements a generalized principal components analysis (GLM-PCA) for dimension reduction of non-normally distributed data such as counts or binary matrices. (<a href="https://doi.org/10.1186/s13059-019-1861-6">Townes et al., 2019. <em>Genome Biology</em></a>).</li>
</ul>
</section>
</section>
<section id="python" class="level2">
<h2 class="anchored" data-anchor-id="python">Python</h2>
<section id="software-packages-2" class="level4">
<h4 class="anchored" data-anchor-id="software-packages-2">Software packages</h4>
<ul>
<li><a href="https://pypi.org/project/spotsweeper">PyPI/spotsweeper</a> | Python implementation of functions that provides spatially-aware quality control metrics for spatial transcriptomics data. (<a href="https://doi.org/10.1038/s41592-025-02713-3">Totty et al., 2025. <em>Nature Methods</em></a>).</li>
<li><a href="https://pypi.org/project/visiumhd-utils">PyPI/spotsweeper</a> | Tools for pre-processing and visualizing Visium HD spatial transcriptomics data.</li>
<li><a href="https://github.com/chaichontat/samui">Python/samui</a> | A performant web-based interactive visualization tool for spatially-resolved transcriptomics experiments called <a href="https://samuibrowser.com">Samui Browser</a>. (<a href="https://doi.org/10.1017/S2633903X2300017X">Sriworarat et al., 2023. <em>Biological Imaging</em></a>)</li>
<li><a href="https://github.com/dnbaker/minicore">Python/minicore</a> | Fast, generic library for constructing and clustering coresets on graphs, in metric spaces and under non-metric dissimilarity measures for count data and probability distributions. (<a href="https://pubmed.ncbi.nlm.nih.gov/34778889/">Baker et al., 2021. <em>ACM Conference on Bioinformatics</em></a>).</li>
</ul>
</section>
</section>
<section id="github" class="level2">
<h2 class="anchored" data-anchor-id="github">GitHub</h2>
<section id="software-packages-3" class="level4">
<h4 class="anchored" data-anchor-id="software-packages-3">Software packages</h4>
<ul>
<li><a href="https://github.com/Pixel-Dream/spatialMNN">R/spatialMNN</a> | Spatially aware unsupervised clustering for spatial omics data. (<a href="https://doi.org/10.1093/bioinformatics/btaf403">Zhou et al.(2025). <em>Bioinformatics</em></a>).</li>
<li><a href="https://github.com/stephaniehicks/quantroSim">R/quantroSim</a> | Supporting data simulation R-package for the <em>quantro</em> R-package to simulate gene expression and DNA methylation data.</li>
<li><a href="https://github.com/hilaryparker/explainr">R/explainr</a> | Translates S3 objects into text using standard templates in a simple and convenient way.</li>
<li><a href="https://github.com/stephaniehicks/postMUT">postMUT</a> | A tool implemented in Perl and R to predict the functionality of missense mutations.</li>
</ul>
</section>
<section id="data-packages-1" class="level4">
<h4 class="anchored" data-anchor-id="data-packages-1">Data packages</h4>
<ul>
<li><a href="https://github.com/stephaniehicks/trapnell2014myoblasthuman">trapnell2014myoblasthuman</a> | R data package that contains an ExpressionSet object from Trapnell et al.&nbsp;(2014) that performed a time-series experiment of bulk and single cell RNA-Seq at four time points in differentiated primary human myoblasts.</li>
<li><a href="https://github.com/willtownes/patel2014gliohuman">patel2014gliohuman</a> | R data package that contains a SummarizedExpression object from Patel et al.&nbsp;(2014) with single cell and bulk RNA-Seq data on five human glioblastoma tumors.</li>
<li><a href="https://github.com/genomicsclass/colonCancerWGBS">colonCancerWGBS</a> | Cov files produced from <a href="http://www.bioinformatics.babraham.ac.uk/projects/bismark/">Bismark</a> after mapping six paired tumor-normal WGBS samples from Ziller et al.&nbsp;(2013) PMID: 23925113. Only chr22.</li>
<li><a href="https://github.com/stephaniehicks/mycAffyData">myAffyData</a> | AffyBatch object from an experiment using P493-6 cells expressing low or high levels of c-Myc. Data from Loven et al.&nbsp;(2012) Cell 151: 476-482.</li>
<li><a href="https://github.com/stephaniehicks/BackgroundExperimentYeast">BackgroundExperimentYeast</a>: AffyBatch object from an experiment to measure NSB and optical noise in yeast.</li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/www\.stephaniehicks\.com");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>