<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Talks – Stephanie C. Hicks, PhD</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41afc036f4045670d8a63ab41900e02e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Stephanie C. Hicks, PhD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./projects.html"> 
<span class="menu-text">Research themes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./people.html"> 
<span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./publications.html"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./software.html"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./teaching.html"> 
<span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./talks.html" aria-current="page"> 
<span class="menu-text">Talks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./join.html"> 
<span class="menu-text">Join us</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/stephaniehicks"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/stephaniehicks"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#section" id="toc-section" class="nav-link active" data-scroll-target="#section">2025</a></li>
  <li><a href="#section-1" id="toc-section-1" class="nav-link" data-scroll-target="#section-1">2024</a></li>
  <li><a href="#section-2" id="toc-section-2" class="nav-link" data-scroll-target="#section-2">2023</a></li>
  <li><a href="#section-3" id="toc-section-3" class="nav-link" data-scroll-target="#section-3">2022</a></li>
  <li><a href="#section-4" id="toc-section-4" class="nav-link" data-scroll-target="#section-4">2021</a></li>
  <li><a href="#earlier-years" id="toc-earlier-years" class="nav-link" data-scroll-target="#earlier-years">Earlier years</a>
  <ul class="collapse">
  <li><a href="#section-5" id="toc-section-5" class="nav-link" data-scroll-target="#section-5">2020</a></li>
  <li><a href="#section-6" id="toc-section-6" class="nav-link" data-scroll-target="#section-6">2019</a></li>
  <li><a href="#section-7" id="toc-section-7" class="nav-link" data-scroll-target="#section-7">2018</a></li>
  <li><a href="#section-8" id="toc-section-8" class="nav-link" data-scroll-target="#section-8">2017</a></li>
  <li><a href="#section-9" id="toc-section-9" class="nav-link" data-scroll-target="#section-9">2016</a></li>
  <li><a href="#section-10" id="toc-section-10" class="nav-link" data-scroll-target="#section-10">2015</a></li>
  <li><a href="#section-11" id="toc-section-11" class="nav-link" data-scroll-target="#section-11">2014</a></li>
  <li><a href="#section-12" id="toc-section-12" class="nav-link" data-scroll-target="#section-12">2011</a></li>
  <li><a href="#section-13" id="toc-section-13" class="nav-link" data-scroll-target="#section-13">2010</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Talks</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section">2025</h2>
<ul>
<li>Department of Genetics and Genomic Sciences. Icahn School of Medicine at Mount Sinai. 2025 Jun 4. New York City, NY. USA.</li>
<li>STATGEN Conference. 2025 21-23 May. Minneapolis, MN. USA.</li>
<li>Biology of Genomes. 2025 May 6-10. Cold Spring Harbor Laboratory. USA.</li>
<li>CZI Cell Science 2025 annual meeting. 2025.</li>
<li>NERVSPAN. 2025 Apr 7-9. Bordeaux, France.</li>
</ul>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1">2024</h2>
<ul>
<li>CSHL Biological Data Science. 2024 Nov 13-16. Cold Spring Harbor Laboratory. USA.</li>
<li><strong>Inaugural Visiting Professor Seminar Series (nominated and selected by HMS student body)</strong>. Department of Biomedical Informatics. Harvard Medical School. 2024 Oct 3. Boston, MA. USA.</li>
<li>Kavli NDI. Johns Hopkins University. 2024 Oct 2. Baltimore, MD. USA.</li>
<li>Nebraska Center for the Prevention of Obesity Diseases through Dietary Molecules. University of Nebraska-Lincoln. 2024 Sept 25. Lincoln, NE. USA.</li>
<li><strong>Keynote presentation</strong> at the Workshop on Spatial and Temporal Statistical Modeling in Molecular Biology at Congressi Stefano Franscini at Monte Verità. 2024 Sept 8-13. Ascona, Switzerland.</li>
<li>Invited session on “Building Better Data Analyses: Theory, Methods, and Lessons Learned”. Joint Statistical Meetings. 2024 Aug 3-8. Portland, OR. USA.</li>
<li><strong>Keynote presentation</strong> at R/Medicine 2024. 2024 Jun 13.</li>
<li>Boston University Bioinformatics Student-Organized Symposium. 2024 June 5. Boston, MA. USA.</li>
<li>Department of Biostatistics, Epidemiology and Informatics. Perelman School of Medicine at the University of Pennsylvania. 2024 May 7. Philadelphia, PA. USA.</li>
<li>Invited session on “Spatial mapping of tissues at single cell resolution: statistical models and computational paradigms”. STATGEN Conference. 2024 1-3 May. Pittsburgh, PA. USA.</li>
<li>Department of Biostatistics. Unviersity of Washington. 2024 Apr 25. Seattle, WA. USA.</li>
<li>Invited session on “Statistical methods for population scale single cell RNA-seq data and spatial transcriptomics”. ENAR Conference. 2024 Mar 10-13. Baltimore, MD. USA.</li>
<li>Co-organizer of Statistical Aspects of Trustworthy Machine Learning at Banff International Research Station. 2024 Feb 11-16. Banff, Alberta. CA.</li>
<li><strong>Plenary talk</strong> at Advances in Genome Biology and Technology (AGBT). 2024 Feb 5-8. Orlando, FL. USA.</li>
<li>Department of Biostatistics. University of Pittsburgh. 2024 Jan 25. Pittsburgh, PA. USA.</li>
<li>Marie Skłodowska-Curie Actions Doctoral Network. 2024 Jan 24. Bordeaux, France.</li>
</ul>
</section>
<section id="section-2" class="level2">
<h2 class="anchored" data-anchor-id="section-2">2023</h2>
<ul>
<li>Invited panel at American College of Neuropsychopharmacology (ACNP) 62nd Annual Meeting. 2023 Dec 3-6. Tampa, FL. USA.</li>
<li>CSHL Computational Genomics Course. 2023 Nov 29-Dec 6. Cold Spring Harbor Laboratory. USA.</li>
<li>Department of Statistical Sciences. Wake Forest University. 2023 Oct 17. Winston-Salem, NC. USA</li>
<li><strong>Keynote presentation</strong> at the Workshop on Reading, Assembling, Analyzing, and Designing Genomic Data at the AI in Health Conference. 2023 Oct 9. Houston, TX. USA.</li>
<li>Myrto Lefkopoulou Distinguished Lecturership. Harvard T. H. Chan School of Public Health. 2023 Sept 21. Boston, MA. USA.</li>
</ul>
<p align="center">
<img src="images/stephanie_myrto.png" alt="2023 Mytro Award" style="width:50%">
</p>
<ul>
<li>Mathematical Institute for Data Science (MINDS) / Center for Imaging Science (CIS). Johns Hopkins University. 2023 Sept 12. Baltimore, MD. USA.</li>
<li>Invited session on “Advanced Statistical Methods in Spatial biology”. Joint Statistical Meetings. 2023 Aug 5-10. Toronto, Ontario. CA.</li>
<li>Reproducible and FAIR Bioinformatics Analysis of Omics Data. 2023 Jul 18. Bar Harbor, ME. USA.</li>
<li>Annual Liverpool School of Tropical Medicine (LSTM) Postgraduate Researcher Conference. 2023 Jun 27-28. Liverpool, UK.</li>
<li>Department of Epidemiology and Biostatistics at Memorial Sloan-Kettering Cancer Center. 2023 June 7. New York City, NY. USA.</li>
<li>Symposium on Epigenome Sciences. Johns Hopkins University. 2023 May 22. Baltimore, MD. USA.</li>
<li>Translational Data Science Integrated Research Center. Fred Hutch. 2023 May 4. Seattle, WA. USA.</li>
<li>Department of Biomedical Engineering. Johns Hopkins University. 2023 Apr 20. Baltimore, MD. USA.</li>
<li>Department of Biostatistics. Vanderbilt University. 2023 Mar 29. Nashville, TN. USA.</li>
<li>Invited session on “Decomposing Admixed Genomics Data: Cell-type-aware Analysis Methodology Advances”. ENAR Conference. 2023 Mar 19-22. Nashville, TN. USA.</li>
<li>Moderator at the ‘Single Cells, Single Cells, Single All the Way’ session. American Association for the Advancement of Science (AAAS) Annual Meeting. 2023 Mar 2-5. Washington D.C. USA.</li>
<li>JHU Biostatistics PhD Visitors Weekend. 2023 Feb 17. Baltimore, MD. USA.</li>
<li>Genetics, Bioinformatics and Systems Biology Colloquium. UCSD Bioinformatics and Systems Biology (BISB) Graduate Program. 2023 Feb 16. San Diego, CA. USA.</li>
<li>Center for Bioinformatics and Computational Biology (CBCB). University of Maryland. 2023 Feb 2. College Park, MD. USA.</li>
<li>Institute for Data Science and Artificial Intelligence. Department of Mathematics. University of Exeter. 2023 Feb 1. Exeter, South West England, United Kingdom.</li>
<li>Boston R/Bioconductor for Genomics. 2023 Jan 27. Boston, MA. USA.</li>
</ul>
</section>
<section id="section-3" class="level2">
<h2 class="anchored" data-anchor-id="section-3">2022</h2>
<ul>
<li>CSHL Winter Course on Computational Genomics Course. 2022 Nov 30-Dec 12. Cold Spring Harbor Laboratory. USA.</li>
<li><a href="https://meetings.cshl.edu/meetings.aspx?meet=DATA&amp;year=22">Biological Data Science</a>. 2022 Nov 9-12. Cold Spring Harbor Laboratory. USA.</li>
<li><strong>Keynote presentation</strong> at the Johns Hopkins Institute for Data Intensive Engeering and Science (IDIES) <a href="https://idies.jhu.edu/news-events/events/idies-annual-symposium/">2022 Annual Symposium</a>. 2022 Oct 20. Baltimore, MD. USA.</li>
<li>Department of Statistics and Data Science. The Wharton School at The University of Pennsylvania. 2022 Oct 19. Philadelphia, PA. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1Y3EEcUU0aj1uVHptO8VS6Rt6j0GZS6j3F5sn4aeuLdM/edit?usp=sharing">Refining best practices for reproducibility to improve the quality of data analyses in statistics and data science</a>. Invited session on “A Multi-disciplinary View of Reproducibility”. <a href="https://ww2.amstat.org/meetings/jsm/2022/">Joint Statistical Meetings</a>. 2022 Aug 6-11. Washington DC. USA.</li>
<li>Increasing the reproducibility and rigor of single-cell RNA-seq through the use of statistics and data science. <a href="https://mdibl.org/course/fair-2022">Reproducible and FAIR Bioinformatics Analysis of Omics Data</a>. 2022 Jul 18. Bar Harbor, ME. USA.</li>
<li>CSHL Summer Course on Single Cell Analysis. 2022 Jul 5-6. Cold Spring Harbor Laboratory. USA.</li>
<li>CSHL Summer Course on Statistical Analysis of Genome Scale Data. 2022 Jul 1-14. Cold Spring Harbor Laboratory. USA.</li>
<li>NHGRI-NIEHS Workshop on RNomics. 2022 24-26 May.</li>
<li>Department of Genetic Medicine. Johns Hopkins University SOM. 2022 May 18. Baltimore, MD. USA.</li>
<li><a href="https://github.com/rladies-eastlansing/2022-design_principles">Design principles for data analysis</a>. <a href="https://www.meetup.com/rladies-east-lansing/events/284104809">R-Ladies East Lansing</a>. 2022 Apr 14.</li>
<li>Department of Biochemistry and Molecular Biology. Michigan State University. 2022 Apr 14. Lansing, MI. USA.</li>
<li>2022 NHGRI Research Training &amp; Career Development Annual Meeting. 2022 Apr 3-6.</li>
<li>Making genomic data science accessible, reproducible, and scalable to improve human health. Department of Biostatistics at Brown School of Public Health. 2022 Mar 23. Providence, RI. USA.</li>
<li>Department of Physiology and Biophysics and the Institute for Computational Biomedicine (ICB) at Weill Cornell Medicine (WCM). 2022 Mar 14. New York City, NY. USA.</li>
<li>Applied Bioinformatics in Life Sciences (4th edition). <a href="https://www.vibconferences.be/events/applied-bioinformatics-in-life-sciences-3rd-edition">VIB Conference</a> (Plenary session). 2022 Mar 10-11. Leuven, Belgium.</li>
<li><a href="https://math.mit.edu/compbiosem/spring21/index.html">Spring MIT Bioinformatics Seminar Series</a>. MIT Math Department and the Computer Science and Artificial Intelligence Laboratory (CSAIL). Massachusetts Institute of Technology. 2022 Feb 23. Boston, MA. USA.</li>
<li>Department of Biostatistics. Virginia Commonwealth University. 2022 Feb 4. Richmond, VA. USA.</li>
<li><a href="https://www.math.mcmaster.ca/news-events/events/g-1-0fqnlmh4j70c0ruepkgssq7rb8_202201282030.html">Department of Mathematics &amp; Statistics. McMaster University</a>. 2022 Jan 28. Hamilton, Ontario, Canada.</li>
</ul>
</section>
<section id="section-4" class="level2">
<h2 class="anchored" data-anchor-id="section-4">2021</h2>
<ul>
<li>Department of Biostatistics. University of Michigan. 2021 Dec 9. Ann Arbor, MI. USA.</li>
<li>AI &amp; Health Data Issues panel discussion at the <a href="https://pheedloop.com/aihealth/site/home/">Stanford AI + Health Online Conference</a>. Institute for Human-Centered Artificial Intelligence at Stanford University and the Stanford Center for Continuing Medical Education (Stanford CME). 2021 Dec 8-9.</li>
<li>Department of Biostatistics, Epidemiology and Informatics. University of Pennsylvania Perelman School of Medicine. 2021 Nov 30. Phildelphia, PA. USA.</li>
<li>6th Annual <a href="https://www.midatlanticbioinformatics.org/agenda">Mid Atlantic Bioinformatics Conference</a>. 2021 Nov 8. Philadelphia, PA. USA. [<a href="https://www.youtube.com/watch?v=ZJF67I7RF0I">Watch on YouTube</a>].</li>
<li>Department of Applied Mathematics and Statistics. Johns Hopkins University. 2021 Nov 4. Baltimore, MD. USA.</li>
<li>Lieber Institute Speaker Series. Lieber Institute for Brain Development. 2021 Oct 26. Baltimore, MD. USA.</li>
<li>Joint Carnegie Mellon University - University of Pittsburgh PhD Program in Computational Biology. 2021 Oct 8. Pittsburgh, PA. USA.</li>
<li>Community of Scholars (COS) Seminar Series in the Clinical and Translational Science Institute (CTSI) K Scholar Program. University at Buffalo. 2021 Sept 27. Buffalo, NY. USA.</li>
<li><a href="https://magazine.amstat.org/blog/2021/05/01/2021-copss-award-winners/">COPSS Leadership Academy</a> Award from the American Statistical Association (ASA) presented at the COPSS Distinguished Achievement Award and Lectureship. <a href="https://ww2.amstat.org/meetings/jsm/2021/">Joint Statistical Meetings</a>. 2021 Aug 7-12. Seattle, WA. USA.</li>
<li>Evaluating the Alignment of a Data Analysis between Analyst and Audience. Invited session on “Theoretical and Empirical Contributions to Data Analytic Practice”. <a href="https://ww2.amstat.org/meetings/jsm/2021/">Joint Statistical Meetings</a>. 2021 Aug 7-12. Seattle, WA. USA.</li>
<li><a href="https://docs.google.com/presentation/d/19RVSt4uHx7r1i1TOvnAW0khzQgdmHa1Pq3U30emIJ1o/edit?usp=sharing">Increasing the reproducibility and rigor of single-cell RNA-seq through the use of statistics and data science</a>. Reproducible and FAIR Bioinformatics Analysis of Omics Data. 2021 Jul 15. Bar Harbor, ME. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1R06CRYVkJ_Fd8k2mXWbsipW3U5XtbOjRBPJAxXV1mYs/edit?usp=sharing">Scalable statistical methods and software for single-cell data science</a>. Maths in the Life Sciences Seminar Series. Joint seminar held between the Applied Maths and Biostatistics and Data Science Departments. University of Manchester. 2021 Jul 12. Manchester, England, UK. [<a href="https://www.youtube.com/watch?v=ldlqaNMtNYM">Watch on YouTube</a>].</li>
<li><a href="https://docs.google.com/presentation/d/1D-0P-Xm4qAnoX_oJs42CankZwbpvGjgq-HESNnq4Y9k/edit?usp=sharing">Scalable statistical methods and software for single-cell data science</a>. Dynamics and Networks in Single-cell Biology Minisymposium at the <a href="https://www.smb2021.org">Society for Mathematical Biology</a>. University of California Riverside. 2021 Jun 13-17. Riverside, CA. USA.</li>
<li><a href="https://docs.google.com/presentation/d/162fFeEOn7cjh7j7JIzXhNVrgeYP0zivE6Fdlu5g_yh0/edit?usp=sharing">Scalable statistical methods and software for single-cell data science</a>. Biostatistics Seminar. Department of Biomedical Data Science. Stanford School of Medicine. 2021 Mar 11. Stanford, CA. USA. [<a href="https://stanford.zoom.us/rec/share/s5WC407FBaSupdzQY1BKYQXsjiNyhQ25mAPHP5P7jC1qERyYuJ-VAbqH1bOeHkQ5.ekxvZh3ymtDU-x51">Watch online</a>].</li>
<li><a href="https://docs.google.com/presentation/d/1F8-zp8FEjQ8xWpWulSEYitN9NcjEBXRoDH4X98xvxHY/edit?usp=sharing">Scalable statistical methods and software for single-cell data science</a>. Stanford Center for Biomedical Informatics Research. Stanford School of Medicine. 2021 Feb 25. Stanford, CA. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1ueWL8XDlifDTsPlAG8lhPx37zzm1YlGKeaNu6gf8CXk/edit?usp=sharing">Addressing Open Challenges in Data Science Education</a>. Department of Statistics, University of British Columbia. 2021 Feb 9. Vancouver, BC. Canada.</li>
<li><a href="https://docs.google.com/presentation/d/1pwYvydQAfQd8V5p2Hxlc8iyWzJ_lUBpo91eL34KeGO0/edit?usp=sharing">Scalable statistical methods and software for single-cell data science</a>. National Cancer Institute (NCI) / Center for Cancer Research (CCR) Bioinformatics Training and Education Program (BTEP) <a href="https://btep.ccr.cancer.gov/seminar-series/seminar/">Distinguished Speakers Seminar Series</a>. 2021 Jan 28.</li>
<li><a href="https://docs.google.com/presentation/d/1gYQobVpbc4idyXzhcxkxoWD3aqR7-Kz5oW3FM69PwFA/edit?usp=sharing">Scalable statistical methods and software for single-cell data science</a>. Department of Bioinformatics and Biostatistics. University of Louisville. 2021 Jan 15. Louisville, KY. USA.</li>
</ul>
</section>
<section id="earlier-years" class="level1">
<h1>Earlier years</h1>
<section id="section-5" class="level2">
<h2 class="anchored" data-anchor-id="section-5">2020</h2>
<ul>
<li><a href="https://americanhealth.jhu.edu/2020-bloomberg-american-health-summit">Bloomberg American Health Summit 2020</a>. 2020 Dec 9. Baltimore, MD. USA.</li>
</ul>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/DgzBSOY5Yc8" frameborder="0" allowfullscreen="">
</iframe>
</p>
<ul>
<li>CZI Essential Open Source Software (EOSS) Annual Meeting. 2020 Dec 8-10.</li>
<li><a href="https://www.stephaniehicks.com/biocdemo/">Large-scale clustering in Bioconductor</a>. CZI Seed Network Annual Meeting. 2020 Nov 17-19.</li>
<li><a href="https://docs.google.com/presentation/d/1PFx_wn4322FuDWzA14jPGZ786FH0jDsqe1NabcfjS6E/edit?usp=sharing">Scaling up single-cell data analyses in Bioconductor</a>. <a href="https://www.singlecells.org.au/webinar-episodes">Oz Single Cell</a>. 2020 Nov 18. Melbourne, Australia.</li>
</ul>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/poEDERGXrQw" frameborder="0" allowfullscreen="">
</iframe>
</p>
<ul>
<li><a href="https://docs.google.com/presentation/d/1rHnFEjBUt9kqWfVfD5X8dOXSFUp3pFmWZoKCni3GATo/edit?usp=sharing">Scaling up single-cell data analyses in Bioconductor</a>. <a href="https://meetings.cshl.edu/meetings.aspx?meet=DATA&amp;year=20">Biological Data Science</a>. 2020 Nov 4-7. Cold Spring Harbor Laboratory. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1DpeB2LgLdzDDobR8_a3QaowRYkW7_OORnhXd-wdYdpg/edit?usp=sharing">More (single-cell) data, more (statistical) problems</a>. Human Medical Genetics and Genomics Program, University of Colorado (Anschutz Medical Campus). 2020 Oct 22. Denver, CO. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1odVZuin9u5iLoW5f3lcjet5WknQr0BwjuH8nrKHY7CM/edit?usp=sharing">Introduction to GenomicRanges and plyranges</a>. <a href="https://www.meetup.com/rladies-tunis/events/273485109/">R-Ladies Tunis</a>. 2020 Oct 6.</li>
</ul>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/TaEsonYBGlE" frameborder="0" allowfullscreen="">
</iframe>
</p>
<ul>
<li><a href="https://docs.google.com/presentation/d/1gouLBEOSgtWdnLSjkn1K8H8dWrfJZdwIqXJtAvU1KJg/edit?usp=sharing">Setting the Stage for Reproducibility in Single-Cell Data Science</a>. Rigor and Reproducibilty in Cancer Research. Rogel Cancer Center. University of Michigan. 2020 Sept 24. Ann Arbor, MI, USA.</li>
<li><a href="https://docs.google.com/presentation/d/12JA73ApG0MWTPmb1e37RMOjS3oiA93cR41Sp6_LexFI/edit?usp=sharing">Addressing Public Health Challenges with Statistics, Genomics and Data Science</a>. Annual presentation on my research interests in “Current Topics” courses as part of my participation as a Preceptor in the Department of Biochemistry and Molecular Biology’s NCI-supported PhD training program.</li>
<li>Panelist on <em>Celebrating Florence Nightingale’s Bicentenary</em>. <a href="https://ww2.amstat.org/meetings/jsm/2020/">Joint Statistical Meetings</a>. 2020 Aug 1-6. Philadelphia, PA, USA.</li>
<li><a href="https://docs.google.com/presentation/d/1wSuRsSwI-uEyiPqtXEW12CBwr5qULK6zO7aPHsUH8T8/edit?usp=sharing">Communicating statistics and data science to the masses via Open Case Studies</a>. <a href="https://ww2.amstat.org/meetings/jsm/2020/">Joint Statistical Meetings</a>. 2020 Aug 1-6. Philadelphia, PA, USA.</li>
<li><a href="https://docs.google.com/presentation/d/1eD2jz20aDKo-hKFkbp7BRpqt-2d7cqt9Ff_todrL0Zo/edit?usp=sharing">Bench pressing single-cell RNA-sequencing imputation methods</a>. <a href="http://bioc2020.bioconductor.org">Bioconductor Conference</a>. 2020 Jul 29-31. Boston, MA. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1Qq52SMfRkP_MsvhxqjvJXa-33LXVSmA5ovmnlsyx1iQ/edit?usp=sharing">Analyzing Genomics Data in R with Bioconductor</a>. <a href="https://www.meetup.com/rladies-riverside/">R-Ladies Riverside</a>. 2020 Jul 22.</li>
<li><a href="https://docs.google.com/presentation/d/10lRcBMdZ_lzWWCW46t1rThJ1kVm7Ip0Yx3UIdicUEAg/edit?usp=sharing">Addressing Real-World Challenges through Data Science Education</a>. Academic Data Science Alliance (ADSA) Data Science Education special interest group (SIG). 2020 Jul 2.</li>
<li>Annual Symposium of the Division of Gastroenterology at Johns Hopkins University. 2020 Jun 23. Baltimore, MD. USA. (Postponed to due COVID-19).</li>
<li><a href="http://www.birs.ca/events/2020/5-day-workshops/20w5197">Mathematical Frameworks for Integrative Analysis of Emerging Biological Data Types</a> at Banff International Research Station. 2020 Jun 14-19.</li>
<li><a href="https://docs.google.com/presentation/d/1TRmPRv4k4aceJbdVQDub4hCXYvIdTeQfDJZfpBu28PQ/edit?usp=sharing">Using R and data science to improve human health</a>. <strong>Keynote presentation</strong> at <a href="https://2020.erum.io">The European R Users Meeting</a>. 2020 June 17-20. Milan, Italy.</li>
</ul>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/YqLb6gCyUGw" frameborder="0" allowfullscreen="">
</iframe>
</p>
<p>Watch a pre-conference interview here:</p>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/8Fy2oKPFeRE" frameborder="0" allowfullscreen="">
</iframe>
</p>
<ul>
<li>Symposium on Data Science and Statistics (SDSS). 2020 Jun 3-5. Pittsburgh, PA. USA.</li>
<li>Fast clustering for single-cell RNA-seq data using mini-batch <em>k</em>-means. <a href="https://enar.org/meetings/spring2020/">ENAR Conference</a>. 2020 Mar 22-25. Nashville, TN. USA.</li>
<li>WiDS Conference 2020. US Bureau of Labor Statistics. 2020 Mar 2. Washington, DC. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1SKY8fEcWmpwaFQCfd2j-Kr7nH4BQ3-Rfhl2WgmPcI3I/edit?usp=sharing">Statistical Genetics and Genomics at Johns Hopkins Biostatistics</a>. Department of Biostatistics, Johns Hopkins Bloomberg School of Public Health. 2020 Feb 21. Baltimore, MD. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1ZSkeK_bnYdVbiqN8gt4gouFeXY91DyKSjI922y3GZFY/edit?usp=sharing">More (single-cell) data, more (statistical) problems</a>. Department of Biostatistics, University of North Carolina Gillings School of Public Health. 2020 Feb 13. Chapel Hill, NC. USA.</li>
<li>Panelist on Data Science Education at the <a href="https://www.mailman.columbia.edu/public-health-now/events/seminars-conferences/data-science-public-health-summit">Data Science for Public Health Summit</a>. 2020 Jan 17. Columbia Mailman School of Public Health. New York City, NY. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1o5dCD3zYh6hOrbD6TxMeKVNV8oYw-SmH8MTKewGIhB4/edit?usp=sharing">More (single-cell) data, more (statistical) problems</a>. Department of Embryology, Carnegie Institution for Science. 2020 Jan 6. Baltimore, MD. USA.</li>
</ul>
</section>
<section id="section-6" class="level2">
<h2 class="anchored" data-anchor-id="section-6">2019</h2>
<ul>
<li><a href="https://normjam.github.io">NormJam</a>: A normalization workshop and jamboree for scRNA-seq data. 2019 Nov 19-20. New York City, NY. USA.</li>
<li><a href="https://sites.google.com/msdse.org/datascienceleadership2019">Data Science Leadership Summit</a>. 2019 Nov 7-9. Santa Fe, NM. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1emK1ORCmRaJLYDYwWPGhaqNRY2D8cdem_EIFmCmqS00/edit?usp=sharing">Addressing Public Health Challenges through Data Science Education</a>. <a href="https://www.apha.org/annualmeeting">American Public Health Association Annual Meeting</a>. 2019 Nov 2-6. Philadelphia, PA. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1whqrmbkilpZopqfg8MQ6MRJR2PYUAAsbCxwK-eq-HDM/edit?usp=sharing">Making data science accessible in the Johns Hopkins Data Science Lab</a>. Mathematics Departments and the Explosive Growth of Computational and Quantitative Offerings in Higher Education. <a href="http://www.ams.org/meetings/coeminiconference">AMS Annual Mini-conference on Education</a>. 2019 Oct 25. Washington, D.C. USA.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/teaching-data-science-through-case-studies-in-public-health">Teaching data science through case studies in Public Health</a>. Joint Statistical Meeings. 2019 Jul 27-Aug 1. Denver, CO. USA.</li>
<li><a href="https://docs.google.com/presentation/d/1AlwLGTlc3ZFxY8PLpCZ5cwD3ktQyQqb1V6qub2I8360/edit?usp=sharing"><code>mbkmeans</code>: fast clustering for single cell data using mini-batch <em>k</em>-means</a>. Bioconductor Conference. 2019 Jul 24-27. New York City, NY. USA.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/why-statistics-matters-in-the-analysis-of-genomics-data">Why Statistics Matters: Analysis of Genomics Data</a>. <a href="http://genomics.jhu.edu/workshop.html">Practical Genomics Workshop</a> at Johns Hopkins University. 2019 June 4-7. Baltimore, MD. USA.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/useful-tools-for-teaching-and-outreach-in-data-science">Useful Tools for Teaching And Outreach In Data Science</a>. Symposium on Data Science and Statistics (SDSS). 2019 May 29-Jun 1. Bellevue, WA. USA.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/addressing-open-challenges-in-data-science-education">Addressing Open Challenges Data Science Education</a>. Department of Statistics and Data Sciences, Smith College. 2019 Apr 18. Northampton, MA. USA.</li>
<li>Missing Data and Technical Variability in Single-Cell RNA-Sequencing Experiments. ENAR Conference. 2019 Mar 24-27. Philadelphia, PA. USA.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/making-data-science-accessible-in-the-johns-hopkins-data-science-lab">Making data science accessible world-wide in the Johns Hopkins Data Science Lab</a>. Department of Statistics, University of Connecticut. 2019 Feb 27. Storrs, CT. USA.</li>
<li>Applications of Latent Variables in Identifying Systematic Errors in Genomics. Department of Statistics, Rice University. 2019 Feb 11. Houston, TX. USA.</li>
<li><a href="https://www.genome.gov/27572552/from-genome-to-phenotype--genomic-variation-identification-association-and-function-in-human-health-and-disease/">National Human Genome Research Institute’s Genome to Phenotype Strategic Planning Meeting</a>. 2019 Jan 22-24. Rockville, MD, USA</li>
<li><a href="https://speakerdeck.com/stephaniehicks/methylcc-presentation">Estimating cell type composition in whole blood using differentially methylated regions</a>. Department of Statistics, Oregon State University. 2019 Jan 14. Corvallis, OR. USA.</li>
</ul>
</section>
<section id="section-7" class="level2">
<h2 class="anchored" data-anchor-id="section-7">2018</h2>
<ul>
<li><a href="https://speakerdeck.com/stephaniehicks/orchestrating-single-cell-rna-sequencing-analysis-with-bioconductor">Orchestrating Single-Cell RNA-sequencing Analysis with Bioconductor</a>.
<ul>
<li><strong>Keynote presentation</strong> at <a href="https://bioconductor.github.io/BiocAsia/">Bioconductor Asia</a> on 2018 Nov 29 at the University of Melbourne, Melbourne, Australia.</li>
<li><strong>Keynote presentation</strong> at <a href="https://bis.amsi.org.au">Australian Mathematical Sciences Institute (AMSI) Bioinfosummer</a> on 2018 Dec 3-7 at the University of Western Australia, Perth, Australia.</li>
</ul></li>
<li><a href="https://speakerdeck.com/stephaniehicks/analyzing-genomics-data-in-r-with-bioconductor">Analyzing Genomics Data in R with Bioconductor</a> at DC R Conference. 2018 Nov 8. Washington D.C., USA.</li>
</ul>
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/l1MQ7x8cn7Y" frameborder="0" allowfullscreen="">
</iframe>
</p>
<ul>
<li><a href="https://speakerdeck.com/stephaniehicks/methylcc-presentation">Estimating cell type composition in whole blood using differentially methylated regions</a>. Department of Biostatistics, The University of Alabama at Birmingham. 2018 Nov 2. Birmingham, AL. USA.</li>
<li>Estimating cell type composition in whole blood using differentially methylated regions at the <a href="http://genomics.jhu.edu/symposium.html">12th Annual Symposium and Poster Session on Genomics and Bioinformatics at Johns Hopkins University</a>. 2018 Oct 25. Baltimore, MD. USA.</li>
<li><a href="https://ww2.amstat.org/meetings/wsds/2018/">Women in Statistics and Data Science Conference</a>. Lead a project on building a children’s book to highlight women in Statistics and Data Science.</li>
<li><a href="https://bigdatau.ini.usc.edu/innovationlab2018">Data Science Innovation Lab: Mathematical Challenges of Single Cell Dynamics</a>. Invited to participate in a five-day workshop with early-career biomedical and quantitative investigators to create collaborations, answering important questions in single cell data.</li>
<li><a href="https://ww2.amstat.org/meetings/jsm/2018/onlineprogram/ActivityDetails.cfm?SessionID=215111">Open challenges in Single-Cell RNA-Sequencing</a>. Joint Statistical Meetings. 2018 Jul 29-Aug 2. Vancouver, CA.</li>
<li>Open Challenges in Single-Cell RNA-Sequencing. <a href="http://lorentzcenter.nl/lc/web/2018/986/info.php3?wsid=986&amp;venue=Oort">Single Cell Data Science: Making Sense of Data from Billions of Single Cells</a>. 2018 Jun 4-8. Lorentz Center, Leiden University, Netherlands.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/methylcc-presentation">Technology-independent estimation of cell type composition using differentially methylated regions</a>. Office of Biostatistics Research at NHLBI/NIH. 2018 Apr 24. Bethesda, MD. USA.</li>
</ul>
</section>
<section id="section-8" class="level2">
<h2 class="anchored" data-anchor-id="section-8">2017</h2>
<ul>
<li>Missing Data and Technical Variability in Single-Cell RNA-Sequencing Experiments. <a href="https://ww2.amstat.org/meetings/jsm/2017/">Joint Statistical Meetings</a>. 2017 Jul 29-Aug 3. Baltimore, MD, USA.</li>
<li>Estimating cell type composition in whole blood using differentially methylated regions. <a href="http://bioconductor.org/help/course-materials/2017/BioC2017/">Bioconductor conference</a>. 2017 Jul 26-28. Boston, MA, USA.</li>
<li>Missing Data and Technical Variability in Single Cell RNA-Sequencing Experiments. <a href="https://www.bsse.ethz.ch/cbg/cbg-news/ascona-2017.html">Ascona Workshop 2017: Statistical Challenges in Single Cell Biology</a>. 2017 Apr 30-May 5. Ascona, Switzerland.</li>
<li>Setting the Stage for Reproducibility and Replicability in Science. Brandeis University. 2017 Mar 22. Waltham, MA, USA.</li>
</ul>
</section>
<section id="section-9" class="level2">
<h2 class="anchored" data-anchor-id="section-9">2016</h2>
<ul>
<li><a href="https://speakerdeck.com/stephaniehicks/towards-progress-in-batch-effects-and-biases-in-single-cell-rna-seq-data">On the widespread and critical impact of systemic bias and batch effects in single-cell RNA-seq data</a>. <a href="http://hsci.harvard.edu/event/widespread-and-critical-impact-systemic-bias-and-batch-effects-single-cell-rna-seq-data?delta=0">Presented</a> at the <a href="http://www.singlecellnetwork.org">Boston Single-Cell Network Meeting</a> in March 2016 (Boston, MA, USA), <a href="https://www.amstat.org/meetings/jsm/2016/onlineprogram/MainSearchResults.cfm">presented</a> at the Joint Statistical Meetings Aug 2016 (Chicago, IL, USA), and <a href="https://coursesandconferences.wellcomegenomecampus.org/events/item.aspx?e=596">presented</a> at the Single-Cell Genomics Conference Sept 2016 (Hinxton, UK).</li>
<li><a href="https://github.com/hms-dbmi/scw/tree/master/scw2016/tutorials/batcheffects">Batch effects and technical biases in scRNA-Seq data</a>. <a href="http://hsci.harvard.edu/event/save-date-single-cell-analysis-workshop">HSCI Single-Cell Workshop</a>. 2016 Nov 29-30. Harvard Medical School, Boston, MA, USA.</li>
<li>Transforming the Classroom to Teach Statistics and Data Science with Active Learning. <a href="http://www.amstat.org/meetings/wsds/2016/">Women in Statistics and Data Science Conference</a>. 2016 Oct 20-22. Charlotte, NC, USA.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/towards-progress-in-batch-effects-and-biases-in-single-cell-rna-seq-data">Towards progress in batch effects and biases in single-cell RNA-seq data</a>. 2016 Sept 14-16. Wellcome Genome Campus, Hinxton, Cambridge, UK.</li>
<li>On the widespread and critical impact of systemic bias and batch effects in single-cell RNA-seq data. <a href="https://www.amstat.org/meetings/jsm/2016/onlineprogram/MainSearchResults.cfm">Joint Statistical Meetings</a>. 2016 Jul 31-Aug 4. Chicago, IL, USA.</li>
<li>On the widespread and critical impact of systemic bias and batch effects in single-cell RNA-seq data. <a href="http://hsci.harvard.edu/event/widespread-and-critical-impact-systemic-bias-and-batch-effects-single-cell-rna-seq-data?delta=0">Boston Single-Cell Network Meeting</a>. 2016 Mar 15. Boston, MA, USA.</li>
</ul>
</section>
<section id="section-10" class="level2">
<h2 class="anchored" data-anchor-id="section-10">2015</h2>
<ul>
<li>On the widespread and critical impact of batch effects and systematic bias in single-cell RNA-Seq data (Selected for a Stellar Abstract Award). <a href="https://www.hsph.harvard.edu/2015-pqg-conference/">PQG Conference: Single-cell Genomics: Technology, Analysis, and Applications</a>. 2015 Nov 5-6. Boston, MA, USA.</li>
<li><a href="https://www.mozillascience.org/open-science-summit-2015">Mozilla Open Science Leadership Summit</a>. Invited to work with community leaders furthering open practice and open science through creating community events, tools for collaboration and learning resources.</li>
<li><a href="https://speakerdeck.com/stephaniehicks/why-statistics-matters-in-the-analysis-of-genomics-data">Why Statistics Matters in the Analysis of Genomics Data</a>. Presented at the LSU Computational Biology seminar and the LSUConnect event. 2015 Feb 11. Baton Rouge, LA, USA.
<p align="center">
<iframe width="560" height="315" src="https://www.youtube.com/embed/3EVtJPz4kCI" frameborder="0" allowfullscreen="">
</iframe>
</p></li>
</ul>
</section>
<section id="section-11" class="level2">
<h2 class="anchored" data-anchor-id="section-11">2014</h2>
<ul>
<li>Normalization of DNA methylation and Gene Expression Data in the Context of Global Variation. Bioinformatics Meeting, Division of Immunology, Harvard Medical School. 2014 Sept 18. Boston, MA, USA.</li>
<li>quantro: When should you use quantile normalization?. Flashlight talk at the <a href="https://www.bioconductor.org/help/course-materials/2014/BioC2014/">Bioconductor Conference</a>. 2014 Jul 30-Aug 1. Boston, MA, USA.</li>
<li><a href="http://unconf.ropensci.org">ROpenSci Unconference</a>. Invited to work with over 40 R enthusiasts from industry, academia, non-profits and government on projects supporting open data, open science and data visualization in R. Contributed to R-packages including <a href="https://github.com/hilaryparker/explainr">explainr</a> and <a href="https://github.com/hilaryparker/catsplainr">catsplainr</a>.</li>
</ul>
</section>
<section id="section-12" class="level2">
<h2 class="anchored" data-anchor-id="section-12">2011</h2>
<ul>
<li>Modeling Discovery Of Functional SNPs From Genome Scale Data. JSM 2011. Miami, FL, USA.</li>
</ul>
</section>
<section id="section-13" class="level2">
<h2 class="anchored" data-anchor-id="section-13">2010</h2>
<ul>
<li>Prediction of Missense Mutation Functionality Depends on both the Algorithm and Sequence Alignment Employed. Human Genome Variation Society’s Exploring the Functional Consequences of Genomic Variation Meeting. Washington, D.C., USA.</li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/www\.stephaniehicks\.com");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>